{
 "metadata": {
  "name": "",
  "signature": "sha256:54a0247c1c9155371f9412a802251c12e5d615ebaf723b817b0c79da4598355f"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import numpy as np\n",
      "import matplotlib\n",
      "import matplotlib.pyplot as plt \n",
      "import pandas as pd\n",
      "from pandas.io import sql\n",
      "import sqlite3\n",
      "import sys\n",
      "import os\n",
      "sys.path.append('/quakestor2/llmartin/infectome/infectome-explorer')\n",
      "%matplotlib inline\n",
      "matplotlib.rcParams['savefig.dpi'] = 2 * matplotlib.rcParams['savefig.dpi']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Data preperation - "
     ]
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Process the 16s data - "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "The first table provides a master list of all 4402 16s samples taken. This includes both a sample ID as well as body site and subject ID."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Basic table - \n",
      "\n",
      "ref_file=pd.read_csv(\"/quakestor2/llmartin/infectome/pregnancy/otu_samples.csv\",dtype=str)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Sites samples - \n",
      "\n",
      "bodySitesSampled=ref_file.groupby('BodySite').size()\n",
      "bodySitesSampled.sort(ascending=False)\n",
      "bodySitesSampled"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 4,
       "text": [
        "BodySite\n",
        "Vaginal_Swab       1072\n",
        "Saliva             1033\n",
        "Tooth_Gum           976\n",
        "Stool               887\n",
        "mock                213\n",
        "NTC                  52\n",
        "Back_of_ear_L        51\n",
        "Urine                34\n",
        "Back_of_ear_R        34\n",
        "Elbow_crease_R       29\n",
        "CordBlood_Serum      21\n",
        "dtype: int64"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Subjects sampled - \n",
      "\n",
      "subjectsSampled=ref_file.groupby('SubjectID').size()\n",
      "subjectsSampled.sort(ascending=False)\n",
      "print \"Number of subjects:\"\n",
      "print subjectsSampled[subjectsSampled>1].shape[0]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of subjects:\n",
        "48\n"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "subjectsSampled=ref_file.groupby('SubjectID').size()\n",
      "subjectsSampled.sort()\n",
      "subjectsSampled"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 6,
       "text": [
        "SubjectID\n",
        "mock056      1\n",
        "mock112      1\n",
        "mock111      1\n",
        "mock110      1\n",
        "mock109      1\n",
        "mock108      1\n",
        "mock107      1\n",
        "mock106      1\n",
        "mock105      1\n",
        "mock104      1\n",
        "mock103      1\n",
        "mock102      1\n",
        "mock101      1\n",
        "mock100      1\n",
        "mock099      1\n",
        "...\n",
        "19005        128\n",
        "10047        130\n",
        "10043        134\n",
        "10018        137\n",
        "10045        138\n",
        "10044        139\n",
        "10006        146\n",
        "10034        151\n",
        "10032        152\n",
        "10046        153\n",
        "10028        155\n",
        "10022        156\n",
        "10039        160\n",
        "10023        166\n",
        "10101        206\n",
        "Length: 314, dtype: int64"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Build dictionary between body site and sampleID -\n",
      "\n",
      "sample_to_site_dict=ref_file[['#SampleID','BodySite']]\n",
      "sample_to_site_dict=sample_to_site_dict.set_index('#SampleID',drop=True)\n",
      "\n",
      "# Not sampleID in this file can be parsed into its constituent parts - \n",
      "ref_file['SubjectID']=ref_file['#SampleID'].apply(lambda x:x[0:5])\n",
      "ref_file['Spec_type']=ref_file['#SampleID'].apply(lambda x:x[5:7])\n",
      "ref_file['Gest_week']=ref_file['#SampleID'].apply(lambda x:x[7:9])\n",
      "ref_file['Bank']=ref_file['#SampleID'].apply(lambda x:x[-1])\n",
      "ref_file['Sample_ID']='PR'+ref_file['SubjectID']+'_'+ref_file['Gest_week']\n",
      "ref_file.set_index('#SampleID',inplace=True)\n",
      "\n",
      "sample_to_site_dict.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>BodySite</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>#SampleID</th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1000201108</th>\n",
        "      <td> Vaginal_Swab</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1000501278</th>\n",
        "      <td> Vaginal_Swab</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1000101018</th>\n",
        "      <td> Vaginal_Swab</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "                BodySite\n",
        "#SampleID               \n",
        "1000201108  Vaginal_Swab\n",
        "1000501278  Vaginal_Swab\n",
        "1000101018  Vaginal_Swab"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Build dictionary between OTU_ID and taxonomic tree -\n",
      "\n",
      "def cleanList(l):\n",
      "    try:\n",
      "        name=l[1]\n",
      "    except IndexError:\n",
      "        name=''\n",
      "    return name\n",
      "\n",
      "otu=pd.read_table(\"/quakestor2/llmartin/infectome/pregnancy/File10b_denovo_otu_table_97ID_TAX_ADDED_MANUALLY.txt\",dtype=str)\n",
      "otu_dict=otu[['OTUId','taxonomy']]\n",
      "otu_dict=otu_dict.set_index('OTUId',drop=True)\n",
      "tmp=list(otu_dict['taxonomy'].apply(lambda x: x.split(';')))\n",
      "taxDat=pd.DataFrame(tmp)\n",
      "taxDat.index=otu_dict.index\n",
      "taxDat=taxDat.astype(str)\n",
      "taxDat=taxDat.apply(lambda x: x.str.split('__').apply(cleanList),axis=1)\n",
      "taxDat.columns=[['Kingdom','Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']]\n",
      "taxDat.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Kingdom</th>\n",
        "      <th>Phylum</th>\n",
        "      <th>Class</th>\n",
        "      <th>Order</th>\n",
        "      <th>Family</th>\n",
        "      <th>Genus</th>\n",
        "      <th>Species</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>OTUId</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>OTU_25</th>\n",
        "      <td> Bacteria</td>\n",
        "      <td> Actinobacteria</td>\n",
        "      <td> Actinobacteria</td>\n",
        "      <td> Actinomycetales</td>\n",
        "      <td> Propionibacteriaceae</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td>       acnes</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>OTU_38</th>\n",
        "      <td> Bacteria</td>\n",
        "      <td>     Firmicutes</td>\n",
        "      <td>        Bacilli</td>\n",
        "      <td>      Bacillales</td>\n",
        "      <td>    Staphylococcaceae</td>\n",
        "      <td>    Staphylococcus</td>\n",
        "      <td> epidermidis</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>OTU_233</th>\n",
        "      <td> Bacteria</td>\n",
        "      <td> Actinobacteria</td>\n",
        "      <td> Actinobacteria</td>\n",
        "      <td> Actinomycetales</td>\n",
        "      <td> Propionibacteriaceae</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td>       acnes</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "          Kingdom          Phylum           Class            Order  \\\n",
        "OTUId                                                                \n",
        "OTU_25   Bacteria  Actinobacteria  Actinobacteria  Actinomycetales   \n",
        "OTU_38   Bacteria      Firmicutes         Bacilli       Bacillales   \n",
        "OTU_233  Bacteria  Actinobacteria  Actinobacteria  Actinomycetales   \n",
        "\n",
        "                       Family              Genus      Species  \n",
        "OTUId                                                          \n",
        "OTU_25   Propionibacteriaceae  Propionibacterium        acnes  \n",
        "OTU_38      Staphylococcaceae     Staphylococcus  epidermidis  \n",
        "OTU_233  Propionibacteriaceae  Propionibacterium        acnes  "
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Pull NCBI tax table - \n",
      "\n",
      "FullTaxTable = pd.read_table(\"/quakestor2/kertesz/soup/Data/NCBI/tax/Taxtastic/min.tax.table\",dtype=str)\n",
      "FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "FullTaxTable.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "      <th>parent_id</th>\n",
        "      <th>rank</th>\n",
        "      <th>tax_name</th>\n",
        "      <th>root</th>\n",
        "      <th>below_root</th>\n",
        "      <th>superkingdom</th>\n",
        "      <th>below_superkingdom</th>\n",
        "      <th>below_below_superkingdom</th>\n",
        "      <th>below_below_below_superkingdom</th>\n",
        "      <th>...</th>\n",
        "      <th>below_below_species</th>\n",
        "      <th>below_below_below_species</th>\n",
        "      <th>subspecies</th>\n",
        "      <th>below_subspecies</th>\n",
        "      <th>below_below_subspecies</th>\n",
        "      <th>below_below_below_subspecies</th>\n",
        "      <th>varietas</th>\n",
        "      <th>below_varietas</th>\n",
        "      <th>forma</th>\n",
        "      <th>below_forma</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_id</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
        "      <td>      1</td>\n",
        "      <td>      1</td>\n",
        "      <td>         root</td>\n",
        "      <td>               root</td>\n",
        "      <td> 1</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td>...</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>131567</th>\n",
        "      <td> 131567</td>\n",
        "      <td>      1</td>\n",
        "      <td>   below_root</td>\n",
        "      <td> cellular organisms</td>\n",
        "      <td> 1</td>\n",
        "      <td> 131567</td>\n",
        "      <td>  NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td>...</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2157</th>\n",
        "      <td>   2157</td>\n",
        "      <td> 131567</td>\n",
        "      <td> superkingdom</td>\n",
        "      <td>            Archaea</td>\n",
        "      <td> 1</td>\n",
        "      <td> 131567</td>\n",
        "      <td> 2157</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td>...</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 61 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 9,
       "text": [
        "        tax_id parent_id          rank            tax_name root below_root  \\\n",
        "tax_id                                                                       \n",
        "1            1         1          root                root    1        NaN   \n",
        "131567  131567         1    below_root  cellular organisms    1     131567   \n",
        "2157      2157    131567  superkingdom             Archaea    1     131567   \n",
        "\n",
        "       superkingdom below_superkingdom below_below_superkingdom  \\\n",
        "tax_id                                                            \n",
        "1               NaN                NaN                      NaN   \n",
        "131567          NaN                NaN                      NaN   \n",
        "2157           2157                NaN                      NaN   \n",
        "\n",
        "       below_below_below_superkingdom              ...                \\\n",
        "tax_id                                             ...                 \n",
        "1                                 NaN              ...                 \n",
        "131567                            NaN              ...                 \n",
        "2157                              NaN              ...                 \n",
        "\n",
        "       below_below_species below_below_below_species subspecies  \\\n",
        "tax_id                                                            \n",
        "1                      NaN                       NaN        NaN   \n",
        "131567                 NaN                       NaN        NaN   \n",
        "2157                   NaN                       NaN        NaN   \n",
        "\n",
        "       below_subspecies below_below_subspecies below_below_below_subspecies  \\\n",
        "tax_id                                                                        \n",
        "1                   NaN                    NaN                          NaN   \n",
        "131567              NaN                    NaN                          NaN   \n",
        "2157                NaN                    NaN                          NaN   \n",
        "\n",
        "       varietas below_varietas forma below_forma  \n",
        "tax_id                                            \n",
        "1           NaN            NaN   NaN         NaN  \n",
        "131567      NaN            NaN   NaN         NaN  \n",
        "2157        NaN            NaN   NaN         NaN  \n",
        "\n",
        "[3 rows x 61 columns]"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Merge OTU and taxTable - \n",
      "\n",
      "# Choose the same taxonomic level for OTU and NCBI data - \n",
      "taxLevel_otuTable='Genus'\n",
      "tmp_OTUs=pd.DataFrame(taxDat[taxLevel_otuTable])\n",
      "tmp_OTUs.reset_index(drop=False,inplace=True)\n",
      "\n",
      "taxLevel_taxTable='genus'\n",
      "tmp_ncbiTree=FullTaxTable[FullTaxTable['rank']==taxLevel_taxTable]\n",
      "tmp_ncbiTree=tmp_ncbiTree[['tax_name','tax_id']]\n",
      "\n",
      "# Merge these based upon tax name, giving us a link between OTU_ID and tax_id, which we need later - \n",
      "taxID_to_OTU_mapping=pd.merge(tmp_OTUs,tmp_ncbiTree,left_on=taxLevel_otuTable,right_on=\"tax_name\")\n",
      "taxID_to_OTU_mapping.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>OTUId</th>\n",
        "      <th>Genus</th>\n",
        "      <th>tax_name</th>\n",
        "      <th>tax_id</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0</th>\n",
        "      <td>  OTU_25</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> 1743</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
        "      <td> OTU_233</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> 1743</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2</th>\n",
        "      <td> OTU_962</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> Propionibacterium</td>\n",
        "      <td> 1743</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "     OTUId              Genus           tax_name tax_id\n",
        "0   OTU_25  Propionibacterium  Propionibacterium   1743\n",
        "1  OTU_233  Propionibacterium  Propionibacterium   1743\n",
        "2  OTU_962  Propionibacterium  Propionibacterium   1743"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"Number of unique genus in NCBI data: %s\"%len(list(set(tmp_ncbiTree['tax_name'])))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of unique genus in NCBI data: 1638\n"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"Number of unique genus in OTU data: %s\"%len(list(set(tmp_OTUs['Genus'])))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of unique genus in OTU data: 255\n"
       ]
      }
     ],
     "prompt_number": 20
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# All genus names in OTU data should be captured by NCBI data!\n",
      "print \"Number of genus in OTU data that are not detected in NCBI: %s\"% len(set(tmp_OTUs['Genus'])-set(tmp_ncbiTree['tax_name']))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number ofgenus in OTU data that are not detected in NCBI: 73\n"
       ]
      }
     ],
     "prompt_number": 22
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"Unique OTUs: %s\"%len(list(set(taxID_to_OTU_mapping['OTUId'])))\n",
      "print \"Unique genus level taxIDs: %s\"%len(list(set(taxID_to_OTU_mapping['tax_id'])))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Unique OTUs: 881\n",
        "Unique genus level taxIDs: 182\n"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# --- Note : return to this issue. Not all genus in OTU data can be mapped to NCBI taxID --- "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 24
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Pull and annotate taxID data - \n",
      "\n",
      "# Pull OTU data table, merge with table above, which extracts all GENUS (specified above) and provides associated tax_ID [columns = sample IDs] - \n",
      "otu=pd.read_table(\"/quakestor2/llmartin/infectome/pregnancy/File10b_denovo_otu_table_97ID_TAX_ADDED_MANUALLY.txt\",dtype=str)\n",
      "otu=otu.set_index('OTUId',drop=False) \n",
      "samples_wOTU_andTax=pd.merge(otu,taxID_to_OTU_mapping,left_on='OTUId',right_on=\"OTUId\")\n",
      "\n",
      "# Add body site data for each sample as a row [body site for each OTU sample_ID] - \n",
      "def getBodySite(sampleID):\n",
      "    try:\n",
      "        bs=sample_to_site_dict.loc[sampleID][0]\n",
      "    except KeyError:\n",
      "        bs=''\n",
      "    return bs\n",
      "b_sites=[getBodySite(col) for col in samples_wOTU_andTax.columns]\n",
      "samples_wOTU_andTax.loc['bodysite']=b_sites\n",
      "\n",
      "# Add \"universal\" sampleID (the one we use in cell-free data) to each OTU sample_ID as a row - \n",
      "def getSamID(col):\n",
      "    try:\n",
      "        sampleID=ref_file.loc[col,'Sample_ID']\n",
      "    except KeyError:\n",
      "        sampleID=col\n",
      "    return sampleID\n",
      "sam_names=[getSamID(col) for col in samples_wOTU_andTax.columns]\n",
      "samples_wOTU_andTax.loc['Sample_ID']=sam_names\n",
      "\n",
      "# Transpose so that OTUs are columns and samples are rows - \n",
      "samples_wOTU_andTax_tp=samples_wOTU_andTax.transpose()\n",
      "\n",
      "# Extract taxIDs assocaited with each OTUid - \n",
      "taxID_list=samples_wOTU_andTax_tp.loc['tax_id']\n",
      "\n",
      "# Replace OTU sample_ID with bodysite and universal sample_ID as the row index - \n",
      "samples_wOTU_andTax_tp.set_index(['Sample_ID','bodysite'],drop=False,inplace=True)\n",
      "\n",
      "# Replace OTUs with taxIDs as the column index - \n",
      "samples_wOTU_andTax_tp.columns=taxID_list\n",
      "\n",
      "# Now we have a dataframe of [sample_ID x tax_ID]\n",
      "samples_wOTU_andTax_tp.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "      <th>1743</th>\n",
        "      <th>1279</th>\n",
        "      <th>1743</th>\n",
        "      <th>1743</th>\n",
        "      <th>848</th>\n",
        "      <th>1578</th>\n",
        "      <th>1263</th>\n",
        "      <th>1578</th>\n",
        "      <th>1743</th>\n",
        "      <th>177971</th>\n",
        "      <th>...</th>\n",
        "      <th>264995</th>\n",
        "      <th>375288</th>\n",
        "      <th>838</th>\n",
        "      <th>283168</th>\n",
        "      <th>1263</th>\n",
        "      <th>577309</th>\n",
        "      <th>838</th>\n",
        "      <th>838</th>\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sample_ID</th>\n",
        "      <th>bodysite</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>OTUId</th>\n",
        "      <th></th>\n",
        "      <td> OTU_25</td>\n",
        "      <td> OTU_38</td>\n",
        "      <td> OTU_233</td>\n",
        "      <td> OTU_962</td>\n",
        "      <td> OTU_1330</td>\n",
        "      <td> OTU_3</td>\n",
        "      <td> OTU_153</td>\n",
        "      <td> OTU_4</td>\n",
        "      <td> OTU_1123</td>\n",
        "      <td> OTU_135</td>\n",
        "      <td>...</td>\n",
        "      <td> OTU_1639</td>\n",
        "      <td> OTU_325</td>\n",
        "      <td> OTU_1431</td>\n",
        "      <td> OTU_1892</td>\n",
        "      <td> OTU_1625</td>\n",
        "      <td> OTU_1827</td>\n",
        "      <td> OTU_1704</td>\n",
        "      <td> OTU_1854</td>\n",
        "      <td>              </td>\n",
        "      <td>      OTUId</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PR10013_17</th>\n",
        "      <th>Back_of_ear_L</th>\n",
        "      <td>   1152</td>\n",
        "      <td>    367</td>\n",
        "      <td>       0</td>\n",
        "      <td>      55</td>\n",
        "      <td>        0</td>\n",
        "      <td>     0</td>\n",
        "      <td>       0</td>\n",
        "      <td>     0</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>...</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td> Back_of_ear_L</td>\n",
        "      <td> PR10013_17</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PR10001_02</th>\n",
        "      <th>Back_of_ear_R</th>\n",
        "      <td>    306</td>\n",
        "      <td>   1091</td>\n",
        "      <td>       0</td>\n",
        "      <td>      20</td>\n",
        "      <td>        0</td>\n",
        "      <td>     0</td>\n",
        "      <td>       1</td>\n",
        "      <td>     0</td>\n",
        "      <td>       21</td>\n",
        "      <td>       0</td>\n",
        "      <td>...</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td> Back_of_ear_R</td>\n",
        "      <td> PR10001_02</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 883 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "tax_id                      1743    1279     1743     1743       848   1578  \\\n",
        "Sample_ID  bodysite                                                           \n",
        "OTUId                     OTU_25  OTU_38  OTU_233  OTU_962  OTU_1330  OTU_3   \n",
        "PR10013_17 Back_of_ear_L    1152     367        0       55         0      0   \n",
        "PR10001_02 Back_of_ear_R     306    1091        0       20         0      0   \n",
        "\n",
        "tax_id                       1263   1578      1743   177971   ...     \\\n",
        "Sample_ID  bodysite                                           ...      \n",
        "OTUId                     OTU_153  OTU_4  OTU_1123  OTU_135   ...      \n",
        "PR10013_17 Back_of_ear_L        0      0         0        0   ...      \n",
        "PR10001_02 Back_of_ear_R        1      0        21        0   ...      \n",
        "\n",
        "tax_id                      264995   375288       838    283168      1263  \\\n",
        "Sample_ID  bodysite                                                         \n",
        "OTUId                     OTU_1639  OTU_325  OTU_1431  OTU_1892  OTU_1625   \n",
        "PR10013_17 Back_of_ear_L         0        0         0         0         0   \n",
        "PR10001_02 Back_of_ear_R         0        0         0         0         0   \n",
        "\n",
        "tax_id                      577309       838       838                 \\\n",
        "Sample_ID  bodysite                                                     \n",
        "OTUId                     OTU_1827  OTU_1704  OTU_1854                  \n",
        "PR10013_17 Back_of_ear_L         0         0         0  Back_of_ear_L   \n",
        "PR10001_02 Back_of_ear_R         0         0         0  Back_of_ear_R   \n",
        "\n",
        "tax_id                        tax_id  \n",
        "Sample_ID  bodysite                   \n",
        "OTUId                          OTUId  \n",
        "PR10013_17 Back_of_ear_L  PR10013_17  \n",
        "PR10001_02 Back_of_ear_R  PR10001_02  \n",
        "\n",
        "[3 rows x 883 columns]"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "samples_wOTU_andTax_tp.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "      <th>1743</th>\n",
        "      <th>1279</th>\n",
        "      <th>1743</th>\n",
        "      <th>1743</th>\n",
        "      <th>848</th>\n",
        "      <th>1578</th>\n",
        "      <th>1263</th>\n",
        "      <th>1578</th>\n",
        "      <th>1743</th>\n",
        "      <th>177971</th>\n",
        "      <th>...</th>\n",
        "      <th>264995</th>\n",
        "      <th>375288</th>\n",
        "      <th>838</th>\n",
        "      <th>283168</th>\n",
        "      <th>1263</th>\n",
        "      <th>577309</th>\n",
        "      <th>838</th>\n",
        "      <th>838</th>\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sample_ID</th>\n",
        "      <th>bodysite</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>OTUId</th>\n",
        "      <th></th>\n",
        "      <td> OTU_25</td>\n",
        "      <td> OTU_38</td>\n",
        "      <td> OTU_233</td>\n",
        "      <td> OTU_962</td>\n",
        "      <td> OTU_1330</td>\n",
        "      <td> OTU_3</td>\n",
        "      <td> OTU_153</td>\n",
        "      <td> OTU_4</td>\n",
        "      <td> OTU_1123</td>\n",
        "      <td> OTU_135</td>\n",
        "      <td>...</td>\n",
        "      <td> OTU_1639</td>\n",
        "      <td> OTU_325</td>\n",
        "      <td> OTU_1431</td>\n",
        "      <td> OTU_1892</td>\n",
        "      <td> OTU_1625</td>\n",
        "      <td> OTU_1827</td>\n",
        "      <td> OTU_1704</td>\n",
        "      <td> OTU_1854</td>\n",
        "      <td>              </td>\n",
        "      <td>      OTUId</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PR10013_17</th>\n",
        "      <th>Back_of_ear_L</th>\n",
        "      <td>   1152</td>\n",
        "      <td>    367</td>\n",
        "      <td>       0</td>\n",
        "      <td>      55</td>\n",
        "      <td>        0</td>\n",
        "      <td>     0</td>\n",
        "      <td>       0</td>\n",
        "      <td>     0</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>...</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td> Back_of_ear_L</td>\n",
        "      <td> PR10013_17</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PR10001_02</th>\n",
        "      <th>Back_of_ear_R</th>\n",
        "      <td>    306</td>\n",
        "      <td>   1091</td>\n",
        "      <td>       0</td>\n",
        "      <td>      20</td>\n",
        "      <td>        0</td>\n",
        "      <td>     0</td>\n",
        "      <td>       1</td>\n",
        "      <td>     0</td>\n",
        "      <td>       21</td>\n",
        "      <td>       0</td>\n",
        "      <td>...</td>\n",
        "      <td>        0</td>\n",
        "      <td>       0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td>        0</td>\n",
        "      <td> Back_of_ear_R</td>\n",
        "      <td> PR10001_02</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 883 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 14,
       "text": [
        "tax_id                      1743    1279     1743     1743       848   1578  \\\n",
        "Sample_ID  bodysite                                                           \n",
        "OTUId                     OTU_25  OTU_38  OTU_233  OTU_962  OTU_1330  OTU_3   \n",
        "PR10013_17 Back_of_ear_L    1152     367        0       55         0      0   \n",
        "PR10001_02 Back_of_ear_R     306    1091        0       20         0      0   \n",
        "\n",
        "tax_id                       1263   1578      1743   177971   ...     \\\n",
        "Sample_ID  bodysite                                           ...      \n",
        "OTUId                     OTU_153  OTU_4  OTU_1123  OTU_135   ...      \n",
        "PR10013_17 Back_of_ear_L        0      0         0        0   ...      \n",
        "PR10001_02 Back_of_ear_R        1      0        21        0   ...      \n",
        "\n",
        "tax_id                      264995   375288       838    283168      1263  \\\n",
        "Sample_ID  bodysite                                                         \n",
        "OTUId                     OTU_1639  OTU_325  OTU_1431  OTU_1892  OTU_1625   \n",
        "PR10013_17 Back_of_ear_L         0        0         0         0         0   \n",
        "PR10001_02 Back_of_ear_R         0        0         0         0         0   \n",
        "\n",
        "tax_id                      577309       838       838                 \\\n",
        "Sample_ID  bodysite                                                     \n",
        "OTUId                     OTU_1827  OTU_1704  OTU_1854                  \n",
        "PR10013_17 Back_of_ear_L         0         0         0  Back_of_ear_L   \n",
        "PR10001_02 Back_of_ear_R         0         0         0  Back_of_ear_R   \n",
        "\n",
        "tax_id                        tax_id  \n",
        "Sample_ID  bodysite                   \n",
        "OTUId                          OTUId  \n",
        "PR10013_17 Back_of_ear_L  PR10013_17  \n",
        "PR10001_02 Back_of_ear_R  PR10001_02  \n",
        "\n",
        "[3 rows x 883 columns]"
       ]
      }
     ],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tmp=samples_wOTU_andTax_tp.loc['PR10013_17']\n",
      "tmp"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 48,
       "text": [
        "(5, 883)"
       ]
      }
     ],
     "prompt_number": 48
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Cell-free data - "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# --- Cell-free data ---\n",
      "\n",
      "# --- NOTE: MAKE SURE PACKAGE IS PULLING FROM NCBI VERSION USED TO PROCESS PR DATA (NCBI04) --- \n",
      "# --- FUTURE: EDIT PACKAGE SO THAT DB, PIPE VERSION PASSED AS PARAMS --- \n",
      "import InfectomeExplorer\n",
      "from InfectomeExplorer.data import *\n",
      "print list(set(grammy_df['Patient_ID']))\n",
      "#conn = sqlite3.connect('/quakestor2/kertesz/soup/sql/soup') \n",
      "#pipeline='V2.2.0'\n",
      "#db='NCBIGenomes06'\n",
      "#samples_annot,grammy_df=make_grammy_df(pipeline,db,conn)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[u'L78', u'L79', u'SCScase0026', u'I5', u'L77', u'L74', u'L75', u'SCScase0017', u'SCScase0030', u'SCScase0027', u'I9', u'L73', u'RP12', u'PR10013', u'L9b', u'PR19005', u'PR19009', u'bP1', u'bP3', u'bP2', u'bP4', u'SCScase0036', u'RP60', u'L19', u'L14', u'L15', u'L16', u'L17', u'L10', u'L11', u'L12', u'L13', u'A51', u'EBOLA', u'L85', u'L84', u'L82', u'L81', u'L80', u'CA18', u'P10', u'P11', u'P12', u'P13', u'P14', u'P15', u'P16', u'P18', u'P2', u'P3', u'P1', u'PR10039', u'P7', u'P4', u'P5', u'PR10034', u'P8', u'P9', u'PR10031', u'A31', u'CA12', u'A46', u'A43', u'A40', u'PR10032', u'CA14', u'NEBI6', u'SCScase0046', u'SCScase0028', u'PR10036', u'SCScase0029', u'CA04', u'P26', u'P21', u'P20', u'CA13', u'SCScase0047', u'PR10029', u'SCScase0050', u'PR10024', u'PR10025', u'K2', u'K7', u'K4', u'PR10101', u'A33', u'A32', u'RP53', u'A30', u'A37', u'RP54', u'A35', u'A34', u'RP58', u'L21', u'L20', u'L22', u'L25', u'L24', u'L27', u'L26', u'L29', u'L28', u'A49', u'P34', u'L41', u'L47', u'SCScase0048', u'SCScase0049', u'I15', u'I14', u'L49', u'SCScase0045', u'I11', u'I10', u'I13', u'I12', u'CA10', u'P33', u'SCScase0042', u'A20', u'A21', u'A22', u'A23', u'A24', u'A25', u'A26', u'A27', u'A28', u'L36', u'L34', u'L35', u'L33', u'L30', u'L31', u'L38', u'L50', u'L54', u'CA09', u'L56', u'L57', u'L58', u'L59', u'CA01', u'CA02', u'CA03', u'SCScase0039', u'SCScase0038', u'SCScase0040', u'PR10044', u'SCScase0032', u'PR10046', u'PR10047', u'SCScase0037', u'SCScase0031', u'SCScase0035', u'PR10043', u'A15', u'A14', u'A17', u'A16', u'A11', u'A10', u'A13', u'A12', u'I1', u'I3', u'I2', u'A19', u'I4', u'I7', u'I6', u'A1', u'RP36', u'A3', u'A2', u'A5', u'A4', u'A7', u'A6', u'A9', u'A8', u'SCScase0034', u'SCScase0044', u'L64', u'I8', u'CA17', u'CA16', u'CA15', u'K21', u'L69', u'L68', u'CA11', u'SCScase0043', u'L6', u'L7', u'L4', u'L5', u'L2', u'L3', u'L1', u'SCScase0021', u'SCScase0041', u'SCScase0024', u'SCScase0025', u'L8', u'L9', u'A36']\n"
       ]
      }
     ],
     "prompt_number": 25
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "reload(InfectomeExplorer.data)\n",
      "from InfectomeExplorer.data import *\n",
      "print list(set(grammy_df['Patient_ID']))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[u'PR10013', u'PR10029', u'PR10039', u'PR19005', u'PR10034', u'PR19009', u'PR10036', u'PR10031', u'PR10024', u'PR10025', u'PR10101', u'PR10044', u'PR10046', u'PR10047', u'PR10032', u'PR10043']\n"
       ]
      }
     ],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Pregnancy data\n",
      "study='PR'\n",
      "GRAMMyAnnot=grammy_df[grammy_df['Study']==study]\n",
      "GRAMMyAnnot['SubjectID']=GRAMMyAnnot['Patient_ID'].apply(lambda x:x.split('PR')[1])\n",
      "print GRAMMyAnnot.shape\n",
      "GRAMMyAnnot.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "(6522, 21)\n"
       ]
      },
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Record_ID</th>\n",
        "      <th>Sample_ID</th>\n",
        "      <th>TaxID</th>\n",
        "      <th>Abundance</th>\n",
        "      <th>Error</th>\n",
        "      <th>Database</th>\n",
        "      <th>Pipeline</th>\n",
        "      <th>genome_size</th>\n",
        "      <th>tax_id</th>\n",
        "      <th>tax_name</th>\n",
        "      <th>...</th>\n",
        "      <th>Blast_count_filtered</th>\n",
        "      <th>Chr21_coverage</th>\n",
        "      <th>Patient_ID</th>\n",
        "      <th>DaysPostTransplant</th>\n",
        "      <th>Study</th>\n",
        "      <th>Abundance_Wt</th>\n",
        "      <th>Wt_Sum</th>\n",
        "      <th>grammyHits</th>\n",
        "      <th>Normalized_Abundance</th>\n",
        "      <th>SubjectID</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>192474</th>\n",
        "      <td> PR10024_37c_232721_V2.2.0_NCBIGenomes06</td>\n",
        "      <td> PR10024_37c</td>\n",
        "      <td> 232721</td>\n",
        "      <td> 0.03216</td>\n",
        "      <td> 0.007746</td>\n",
        "      <td> NCBIGenomes06</td>\n",
        "      <td> V2.2.0</td>\n",
        "      <td> 4585154</td>\n",
        "      <td> 232721</td>\n",
        "      <td>                       Acidovorax sp. JS42</td>\n",
        "      <td>...</td>\n",
        "      <td> 64</td>\n",
        "      <td> 0.307146</td>\n",
        "      <td> PR10024</td>\n",
        "      <td> 259</td>\n",
        "      <td> PR</td>\n",
        "      <td> 147458.55264</td>\n",
        "      <td> 4658204.921872</td>\n",
        "      <td> 2.025962</td>\n",
        "      <td> 0.000001</td>\n",
        "      <td> 10024</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>192475</th>\n",
        "      <td> PR10024_37c_866768_V2.2.0_NCBIGenomes06</td>\n",
        "      <td> PR10024_37c</td>\n",
        "      <td> 866768</td>\n",
        "      <td> 0.05769</td>\n",
        "      <td>   0.0278</td>\n",
        "      <td> NCBIGenomes06</td>\n",
        "      <td> V2.2.0</td>\n",
        "      <td> 4570938</td>\n",
        "      <td> 866768</td>\n",
        "      <td> Escherichia coli 'BL21-Gold(DE3)pLysS AG'</td>\n",
        "      <td>...</td>\n",
        "      <td> 64</td>\n",
        "      <td> 0.307146</td>\n",
        "      <td> PR10024</td>\n",
        "      <td> 259</td>\n",
        "      <td> PR</td>\n",
        "      <td> 263697.41322</td>\n",
        "      <td> 4658204.921872</td>\n",
        "      <td> 3.622991</td>\n",
        "      <td> 0.000003</td>\n",
        "      <td> 10024</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>192476</th>\n",
        "      <td> PR10024_37c_335659_V2.2.0_NCBIGenomes06</td>\n",
        "      <td> PR10024_37c</td>\n",
        "      <td> 335659</td>\n",
        "      <td> 0.01740</td>\n",
        "      <td>  0.01018</td>\n",
        "      <td> NCBIGenomes06</td>\n",
        "      <td> V2.2.0</td>\n",
        "      <td> 7231841</td>\n",
        "      <td> 335659</td>\n",
        "      <td>                 Bradyrhizobium sp. S23321</td>\n",
        "      <td>...</td>\n",
        "      <td> 64</td>\n",
        "      <td> 0.307146</td>\n",
        "      <td> PR10024</td>\n",
        "      <td> 259</td>\n",
        "      <td> PR</td>\n",
        "      <td> 125834.03340</td>\n",
        "      <td> 4658204.921872</td>\n",
        "      <td> 1.728859</td>\n",
        "      <td> 0.000001</td>\n",
        "      <td> 10024</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 21 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 26,
       "text": [
        "                                      Record_ID    Sample_ID   TaxID  \\\n",
        "192474  PR10024_37c_232721_V2.2.0_NCBIGenomes06  PR10024_37c  232721   \n",
        "192475  PR10024_37c_866768_V2.2.0_NCBIGenomes06  PR10024_37c  866768   \n",
        "192476  PR10024_37c_335659_V2.2.0_NCBIGenomes06  PR10024_37c  335659   \n",
        "\n",
        "        Abundance     Error       Database Pipeline  genome_size  tax_id  \\\n",
        "192474    0.03216  0.007746  NCBIGenomes06   V2.2.0      4585154  232721   \n",
        "192475    0.05769    0.0278  NCBIGenomes06   V2.2.0      4570938  866768   \n",
        "192476    0.01740   0.01018  NCBIGenomes06   V2.2.0      7231841  335659   \n",
        "\n",
        "                                         tax_name  \\\n",
        "192474                        Acidovorax sp. JS42   \n",
        "192475  Escherichia coli 'BL21-Gold(DE3)pLysS AG'   \n",
        "192476                  Bradyrhizobium sp. S23321   \n",
        "\n",
        "                          ...                     Blast_count_filtered  \\\n",
        "192474                    ...                                       64   \n",
        "192475                    ...                                       64   \n",
        "192476                    ...                                       64   \n",
        "\n",
        "        Chr21_coverage  Patient_ID DaysPostTransplant  Study  Abundance_Wt  \\\n",
        "192474        0.307146     PR10024                259     PR  147458.55264   \n",
        "192475        0.307146     PR10024                259     PR  263697.41322   \n",
        "192476        0.307146     PR10024                259     PR  125834.03340   \n",
        "\n",
        "                Wt_Sum  grammyHits  Normalized_Abundance  SubjectID  \n",
        "192474  4658204.921872    2.025962              0.000001      10024  \n",
        "192475  4658204.921872    3.622991              0.000003      10024  \n",
        "192476  4658204.921872    1.728859              0.000001      10024  \n",
        "\n",
        "[3 rows x 21 columns]"
       ]
      }
     ],
     "prompt_number": 26
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Combine cell-free and 16s - "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Patients for which sample names are not exact matches. \n",
      "# This is because blood is not always drawn at exactly the same time as body site sampled.  \n",
      "No 16s data listed for PR10039_40!\n",
      "No 16s data listed for PR10032_10!\n",
      "No 16s data listed for PR10025_9!\n",
      "No 16s data listed for PR10029_8!\n",
      "No 16s data listed for PR10039_8!\n",
      "No 16s data listed for PR10043_9!\n",
      "No 16s data listed for PR10031_8!\n",
      "No 16s data listed for PR10047_9!\n",
      "No 16s data listed for PR10034_7!\n",
      "No 16s data listed for PR10046_5!\n",
      "No 16s data listed for PR19009_30!\n",
      "No 16s data listed for PR19009_15!\n",
      "No 16s data listed for PR10044_7!\n",
      "No 16s data listed for PR10032_39c!\n",
      "No 16s data listed for PR10031_29c!\n",
      "No 16s data listed for PR10047_39c!\n",
      "No 16s data listed for PR10025_38c!\n",
      "No 16s data listed for PR10034_38c!\n",
      "No 16s data listed for PR10046_39c!\n",
      "No 16s data listed for PR10024_37c!\n",
      "No 16s data listed for PR19005_39c!\n",
      "No 16s data listed for PR10101_36c!\n",
      "No 16s data listed for PR10039_40c!\n",
      "No 16s data listed for PR10043_39c!"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# --- NOTE: Improve matching so that we expand the scope of body site samples! --- \n",
      "sID='10032'\n",
      "tmp=ref_file[ref_file['SubjectID']==sID]\n",
      "tmp.sort('Gest_week',inplace=True,ascending=True)\n",
      "tmp.tail(5)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>BarcodeSequence</th>\n",
        "      <th>LinkerPrimerSequence</th>\n",
        "      <th>BodySite</th>\n",
        "      <th>SubjectID</th>\n",
        "      <th>Run_No</th>\n",
        "      <th>Number_within_run</th>\n",
        "      <th>ReversePrimer</th>\n",
        "      <th>Description</th>\n",
        "      <th>Spec_type</th>\n",
        "      <th>Gest_week</th>\n",
        "      <th>Bank</th>\n",
        "      <th>Sample_ID</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>#SampleID</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1003201748</th>\n",
        "      <td> AGCTATCCACGA</td>\n",
        "      <td> GCACTCCTACGGGAGGCAGCA</td>\n",
        "      <td> Vaginal_Swab</td>\n",
        "      <td> 10032</td>\n",
        "      <td> mod016</td>\n",
        "      <td>  82</td>\n",
        "      <td> CGTATCGCCTCCCTCGCGCCATCAG</td>\n",
        "      <td> 3699</td>\n",
        "      <td> 01</td>\n",
        "      <td> 74</td>\n",
        "      <td> 8</td>\n",
        "      <td> PR10032_74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1003203748</th>\n",
        "      <td> GATCTTCAGTAC</td>\n",
        "      <td> GCACTCCTACGGGAGGCAGCA</td>\n",
        "      <td>       Saliva</td>\n",
        "      <td> 10032</td>\n",
        "      <td> mod017</td>\n",
        "      <td>  55</td>\n",
        "      <td> CGTATCGCCTCCCTCGCGCCATCAG</td>\n",
        "      <td> 3927</td>\n",
        "      <td> 03</td>\n",
        "      <td> 74</td>\n",
        "      <td> 8</td>\n",
        "      <td> PR10032_74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1003204748</th>\n",
        "      <td> TAGACTGTACTC</td>\n",
        "      <td> GCACTCCTACGGGAGGCAGCA</td>\n",
        "      <td>    Tooth_Gum</td>\n",
        "      <td> 10032</td>\n",
        "      <td> mod017</td>\n",
        "      <td> 166</td>\n",
        "      <td> CGTATCGCCTCCCTCGCGCCATCAG</td>\n",
        "      <td> 4038</td>\n",
        "      <td> 04</td>\n",
        "      <td> 74</td>\n",
        "      <td> 8</td>\n",
        "      <td> PR10032_74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1003202748</th>\n",
        "      <td> TCTCACTAGGTA</td>\n",
        "      <td> GCACTCCTACGGGAGGCAGCA</td>\n",
        "      <td>        Stool</td>\n",
        "      <td> 10032</td>\n",
        "      <td> mod017</td>\n",
        "      <td> 222</td>\n",
        "      <td> CGTATCGCCTCCCTCGCGCCATCAG</td>\n",
        "      <td> 4094</td>\n",
        "      <td> 02</td>\n",
        "      <td> 74</td>\n",
        "      <td> 8</td>\n",
        "      <td> PR10032_74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1003202788</th>\n",
        "      <td> TCTCCGCATGTC</td>\n",
        "      <td> GCACTCCTACGGGAGGCAGCA</td>\n",
        "      <td>        Stool</td>\n",
        "      <td> 10032</td>\n",
        "      <td> mod017</td>\n",
        "      <td> 223</td>\n",
        "      <td> CGTATCGCCTCCCTCGCGCCATCAG</td>\n",
        "      <td> 4095</td>\n",
        "      <td> 02</td>\n",
        "      <td> 78</td>\n",
        "      <td> 8</td>\n",
        "      <td> PR10032_78</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 29,
       "text": [
        "           BarcodeSequence   LinkerPrimerSequence      BodySite SubjectID  \\\n",
        "#SampleID                                                                   \n",
        "1003201748    AGCTATCCACGA  GCACTCCTACGGGAGGCAGCA  Vaginal_Swab     10032   \n",
        "1003203748    GATCTTCAGTAC  GCACTCCTACGGGAGGCAGCA        Saliva     10032   \n",
        "1003204748    TAGACTGTACTC  GCACTCCTACGGGAGGCAGCA     Tooth_Gum     10032   \n",
        "1003202748    TCTCACTAGGTA  GCACTCCTACGGGAGGCAGCA         Stool     10032   \n",
        "1003202788    TCTCCGCATGTC  GCACTCCTACGGGAGGCAGCA         Stool     10032   \n",
        "\n",
        "            Run_No Number_within_run              ReversePrimer Description  \\\n",
        "#SampleID                                                                     \n",
        "1003201748  mod016                82  CGTATCGCCTCCCTCGCGCCATCAG        3699   \n",
        "1003203748  mod017                55  CGTATCGCCTCCCTCGCGCCATCAG        3927   \n",
        "1003204748  mod017               166  CGTATCGCCTCCCTCGCGCCATCAG        4038   \n",
        "1003202748  mod017               222  CGTATCGCCTCCCTCGCGCCATCAG        4094   \n",
        "1003202788  mod017               223  CGTATCGCCTCCCTCGCGCCATCAG        4095   \n",
        "\n",
        "           Spec_type Gest_week Bank   Sample_ID  \n",
        "#SampleID                                        \n",
        "1003201748        01        74    8  PR10032_74  \n",
        "1003203748        03        74    8  PR10032_74  \n",
        "1003204748        04        74    8  PR10032_74  \n",
        "1003202748        02        74    8  PR10032_74  \n",
        "1003202788        02        78    8  PR10032_78  "
       ]
      }
     ],
     "prompt_number": 29
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Correct for dislocations between when blood (key) was collected and 16s (value).\n",
      "# For any cfDNA sample [key], this indicates the closest assocaited 16s sample [value]. \n",
      "correction_dict={'PR10039_40':'PR10039_39','PR10032_10':'PR10032_12','PR10025_9':'PR10025_10','PR10029_8':'PR10029_11','PR10039_8':'PR10039_09',\n",
      "                 'PR10043_9':'PR10043_09','PR10031_8':'PR10031_08','PR10047_9':'PR10047_11','PR10034_7':'PR10034_08','PR10046_5':'PR10046_06',\n",
      "                 'PR19009_30':'PR19009_25','PR19009_15':'PR19009_14','PR10044_7':'PR10044_08','PR10032_39c':'PR10032_39','PR10031_29c':'PR10031_29',\n",
      "                 'PR10047_39c':'PR10047_39','PR10025_38c':'PR10025_38','PR10034_38c':'PR10034_38','PR10046_39c':'PR10046_39','PR10024_37c':'PR10024_37',\n",
      "                 'PR19005_39c':'PR19005_39','PR10101_36c':'PR10101_36','PR10039_40c':'PR10039_39','PR10043_39c':'PR10043_39'}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 30
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def get_sample_data(patient_to_analyze):\n",
      "\n",
      "    # --- Get cell-free data for patient --- \n",
      "\n",
      "    # The OTU data is pruned at a specified taxonomic level (specified above).\n",
      "    # Make sure cell-free data is also pruned at this level.\n",
      "    # This is because we need to merge on taxIDs --- so this must be consistent.\n",
      "    level=taxLevel_taxTable \n",
      "    path=\"/quakestor2/kertesz/soup/Data/NCBI/tax/Taxtastic/min.tax.table\"\n",
      "    FullTaxTable=pd.read_table(path,dtype=str)[[\"tax_id\",\"tax_name\",level]]\n",
      "    FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "    mm=pd.merge(GRAMMyAnnot,FullTaxTable,left_on=\"TaxID\",right_on=\"tax_id\")\n",
      "    mm1=mm[['Normalized_Abundance','Sample_ID',level]]\n",
      "    counts=mm1.groupby(['Sample_ID',level],as_index=False).aggregate(np.sum)\n",
      "    counts.set_index('Sample_ID',inplace=True)\n",
      "    \n",
      "    # Slice cell-free data for the patient of interest.\n",
      "    blood_data_toConcat=counts.loc[patient_to_analyze]\n",
      "    scaleFactor=1000000\n",
      "    blood_data_toConcat['Normalized_Abundance']=blood_data_toConcat['Normalized_Abundance']*scaleFactor\n",
      "    blood_data_toConcat.set_index(level,inplace=True)\n",
      "    blood_data_toConcat.columns=['Blood']\n",
      "    \n",
      "    # --- Get 16s data for patient --- \n",
      "    \n",
      "    # 16s data for patient. \n",
      "    try:\n",
      "        patient_16s_data=samples_wOTU_andTax_tp.loc[patient_to_analyze]\n",
      "    except KeyError:\n",
      "        #print \"Correct 16s data for %s\"%patient_to_analyze\n",
      "        patient_16s_data=samples_wOTU_andTax_tp.loc[correction_dict[patient_to_analyze]]\n",
      "            \n",
      "    # List for storage.\n",
      "    storeData=[]\n",
      "    \n",
      "    # Iterate through body sites for the given patient.  \n",
      "    for bodysite in list(set(patient_16s_data.index)):\n",
      "        \n",
      "        # This returns a series indexed by taxID, or a dataframe (if > 1 sample).\n",
      "        patient_16s_data_bodysite=patient_16s_data.loc[bodysite]\n",
      "        \n",
      "        try:\n",
      "            # Sometimes samples will have more than one body site entiry. \n",
      "            # Presumbly two samples taken and sequenced. \n",
      "            # In this case, the Dataframe will have two dimentions (as opposed to a series). \n",
      "            # Iterate through the individual entires and process.\n",
      "            flag=patient_16s_data_bodysite.shape[1]\n",
      "            #print \"Duplicate entries for %s and bodysite %s\"%(patient_to_analyze,bodysite)\n",
      "            \n",
      "            # This stores data from the entries. \n",
      "            tmp_stor=[]\n",
      "            for i in range(patient_16s_data_bodysite.shape[0]):\n",
      "                # This returns a series indexed by taxID.\n",
      "                patient_16s_data_bodysite_dupEntry=patient_16s_data_bodysite.ix[i]\n",
      "                patient_16s_data_bodysite_dupEntry.drop('',inplace=True)\n",
      "                patient_16s_data_bodysite_dupEntry.drop('tax_id',inplace=True)\n",
      "                patient_16s_data_bodysite_dupEntry=patient_16s_data_bodysite_dupEntry.astype(float)\n",
      "                # Aggregate data for multiple instances of a given (genus level) taxID. \n",
      "                patient_16s_data_bodysite_grouped=patient_16s_data_bodysite_dupEntry.groupby(patient_16s_data_bodysite_dupEntry.index).sum()\n",
      "                patient_16s_toConcat=pd.DataFrame(patient_16s_data_bodysite_grouped)\n",
      "                tmp_stor=tmp_stor+[patient_16s_toConcat]\n",
      "            tmp_concat=pd.concat(tmp_stor,axis=1)\n",
      "            \n",
      "            # Take the mean from duplicate body site measurments - \n",
      "            tmp_concat_toConcat=pd.DataFrame(tmp_concat.mean(axis=1))\n",
      "            tmp_concat_toConcat.columns=[bodysite]\n",
      "            #print \"Shape of pruned %s level 16s data: %s\"%(level,tmp_concat_toConcat.shape[0])\n",
      "            \n",
      "            storeData=storeData+[tmp_concat_toConcat]\n",
      "        \n",
      "        # If there is a single body site measurment, Indexerror above is thrown. Process the series in isolation.\n",
      "        except IndexError:\n",
      "            patient_16s_data_bodysite.drop('',inplace=True)\n",
      "            patient_16s_data_bodysite.drop('tax_id',inplace=True)\n",
      "            patient_16s_data_bodysite=patient_16s_data_bodysite.astype(float)\n",
      "            patient_16s_data_bodysite_grouped=patient_16s_data_bodysite.groupby(patient_16s_data_bodysite.index).sum()\n",
      "            patient_16s_toConcat=pd.DataFrame(patient_16s_data_bodysite_grouped)\n",
      "            patient_16s_toConcat.columns=[bodysite]\n",
      "            #print \"Shape of pruned %s level 16s data: %s\"%(level,patient_16s_toConcat.shape[0])\n",
      "            storeData=storeData+[patient_16s_toConcat]\n",
      "    \n",
      "    # List of data from all body sites for samples. Add the corresponding cell-free DNA (blood) data. \n",
      "    storeData=storeData+[blood_data_toConcat]\n",
      "    data_all_cat=pd.concat(storeData,axis=1)\n",
      "    data_all_cat.fillna(0,inplace=True)\n",
      "    data_all_cat_norm=data_all_cat\n",
      "    \n",
      "    # Sort data by load \n",
      "    sums=data_all_cat_norm.sum(axis=1)\n",
      "    sums.sort(ascending=False)\n",
      "    data_all_cat_norm=data_all_cat_norm.loc[sums.index]\n",
      "    #print \"Shape of pruned %s level condolodated blood and 16s data: %s\"%(level,data_all_cat_norm.shape[0])\n",
      "        \n",
      "    # Specify final level of aggregation desired.\n",
      "    # Class is nice beacuse it make viewing the data easier (fewer taxonomic categories) later.\n",
      "    level_to_view_data='genus'\n",
      "    FullTaxTable=pd.read_table(path,dtype=str)[[\"tax_id\",\"tax_name\",level_to_view_data]]\n",
      "    FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "    merge_data=pd.merge(data_all_cat_norm,FullTaxTable,left_on=data_all_cat_norm.index,right_on=\"tax_id\")\n",
      "    merge_data.set_index(['tax_id','tax_name'],inplace=True)\n",
      "    data_abs=merge_data.groupby([level_to_view_data],as_index=False).aggregate(np.sum)\n",
      "    data_abs.set_index(level_to_view_data,inplace=True)\n",
      "    data_abs_annot=pd.merge(data_abs,FullTaxTable,left_on=data_abs.index,right_on=\"tax_id\")\n",
      "    #print \"Shape of final pruned %s level condolodated blood and 16s data: %s\"%(level,data_abs_annot.shape[0])\n",
      "        \n",
      "    return data_abs_annot,level_to_view_data"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 64
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Process matched samples - "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# This will store cfDNA data with matched 16s data --- \n",
      "tmp_store=[]\n",
      "for sample in list(set(GRAMMyAnnot['Sample_ID'])):\n",
      "    \n",
      "    # Cleanup for concatenation\n",
      "    dat_sample,level_used_for_final_prune=get_sample_data(sample)\n",
      "    try:\n",
      "        \n",
      "        # Prepare sample data\n",
      "        dat_sample.set_index(['tax_id','tax_name'],inplace=True)\n",
      "        del dat_sample[level_used_for_final_prune]\n",
      "        dat_sample_norm=dat_sample.div(dat_sample.sum(axis=0),axis=1)\n",
      "        dat_sample_norm.fillna(0,inplace=True)\n",
      "        \n",
      "        # Add sample name to the index \n",
      "        dat_sample_tp=dat_sample.transpose()\n",
      "        dat_sample_tp['Sample']=sample\n",
      "        dat_sample_tp.set_index('Sample',append=True,inplace=True,drop=True)\n",
      "        dat_sample_withName=dat_sample_tp.transpose()\n",
      "        \n",
      "        # Store\n",
      "        tmp_store=tmp_store+[dat_sample_withName]\n",
      "        \n",
      "    except AttributeError:\n",
      "        None\n",
      "        # print \"No 16s data for %s\"%sample\n",
      "        \n",
      "# Concatenate data - \n",
      "data_all_samples_bodysites=pd.concat(tmp_store,axis=1)\n",
      "data_all_samples_bodysites.fillna(0,inplace=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 65
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"Total genus identified: %s\"%len(list(set(data_all_samples_bodysites.index)))\n",
      "print \"Genus identified from 16s data: %s\"%(182)\n",
      "print \"Genus identified exclusivly from metagenomics data: %s\"%(len(list(set(data_all_samples_bodysites.index)))-182)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Total genus identified: 494\n",
        "Genus identified from 16s data: 182\n",
        "Genus identified exclusivly from metagenomics data: 312\n"
       ]
      }
     ],
     "prompt_number": 71
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Save data - \n",
      "print data_all_samples_bodysites.shape\n",
      "data_all_samples_bodysites.columns.names = ['BodySite','Sample']\n",
      "outfilepathToSave=os.getcwd()+'/PR_sample_data_matched_%s.csv'%taxLevel_taxTable\n",
      "data_all_samples_bodysites.to_csv(outfilepathToSave)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "(494, 268)\n"
       ]
      }
     ],
     "prompt_number": 81
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Process all samples - "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# --- This will store all 16s data from the patients in which cfDNA is also sampled --- \n",
      "\n",
      "samples=pd.Series(samples_wOTU_andTax_tp.index.get_level_values('Sample_ID'))\n",
      "sample_IDs=samples[samples.str.contains('PR')]\n",
      "\n",
      "# --- Unique sample names to iterate through ---\n",
      "\n",
      "# Get all 16s samples for patients in which cfDNA was also sampled --- \n",
      "patients_with_cfData=list(set([sam.split('_')[0] for sam in list(set(GRAMMyAnnot['Sample_ID']))]))\n",
      "sample_list=list(samples_wOTU_andTax_tp.index.get_level_values('Sample_ID'))\n",
      "relevant_16s_samples=[sam for sam in sample_list if sam.split('_')[0] in patients_with_cfData]\n",
      "\n",
      "# Combine this with the timepoint for cfDNA samples --- \n",
      "list_of_cfDNA_samples=list(set(GRAMMyAnnot['Sample_ID']))\n",
      "consolodated_samples_for_patients=list(set(list_of_cfDNA_samples+relevant_16s_samples))\n",
      "\n",
      "# The OTU data is pruned at a specified taxonomic level (specified above). Group the cfDNA data at the same level.\n",
      "level=taxLevel_taxTable \n",
      "path=\"/quakestor2/kertesz/soup/Data/NCBI/tax/Taxtastic/min.tax.table\"\n",
      "FullTaxTable=pd.read_table(path,dtype=str)[[\"tax_id\",\"tax_name\",level]]\n",
      "FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "mm=pd.merge(GRAMMyAnnot,FullTaxTable,left_on=\"TaxID\",right_on=\"tax_id\")\n",
      "mm1=mm[['Normalized_Abundance','Sample_ID',level]]\n",
      "counts_cfDNA=mm1.groupby(['Sample_ID',level],as_index=False).aggregate(np.sum)\n",
      "counts_cfDNA.set_index('Sample_ID',inplace=True)\n",
      "\n",
      "# Merge data\n",
      "def merge_16s_with_cfDNA(sample_to_analyze):\n",
      "\n",
      "     # Get cfDNA for sample - \n",
      "    flag_cf=1\n",
      "    try:\n",
      "        blood_data_toConcat=counts_cfDNA.loc[sample_to_analyze]\n",
      "        scaleFactor=1000000\n",
      "        blood_data_toConcat['Normalized_Abundance']=blood_data_toConcat['Normalized_Abundance']*scaleFactor\n",
      "        blood_data_toConcat.set_index(level,inplace=True)\n",
      "        blood_data_toConcat.columns=['Blood']\n",
      "    except:\n",
      "        flag_cf=0\n",
      "        #print \"No cfDNA data for patient: %s\"%sample_to_analyze\n",
      "        blood_data_toConcat=[]\n",
      "        \n",
      "    # Get 16s data for sample - \n",
      "    flag_16s=1\n",
      "    try:\n",
      "        patient_16s_data=samples_wOTU_andTax_tp.loc[sample_to_analyze]\n",
      "    except KeyError:\n",
      "        #print \"Correct 16s data for %s\"%sample_to_analyze\n",
      "        patient_16s_data=[]\n",
      "        flag_16s=0\n",
      "\n",
      "    # List for storage of 16s data - \n",
      "    storeData=[]\n",
      "        \n",
      "    # If there is 16s data, then iterate through body sites - \n",
      "    if flag_16s:\n",
      "    \n",
      "        # Iterate through body sites for the given patient.  \n",
      "        for bodysite in list(set(patient_16s_data.index)):\n",
      "\n",
      "            # This returns a series indexed by taxID, or a dataframe (if > 1 sample).\n",
      "            patient_16s_data_bodysite=patient_16s_data.loc[bodysite]\n",
      "\n",
      "            try:\n",
      "                # Sometimes samples will have more than one body site entiry. \n",
      "                # Presumbly two samples taken and sequenced. \n",
      "                # In this case, the Dataframe will have two dimentions (as opposed to a series). \n",
      "                # Iterate through the individual entires and process.\n",
      "                flag=patient_16s_data_bodysite.shape[1]\n",
      "                # print \"Duplicate entries for %s and bodysite %s\"%(patient_to_analyze,bodysite)\n",
      "\n",
      "                # This stores data from the entries. \n",
      "                tmp_stor=[]\n",
      "                for i in range(patient_16s_data_bodysite.shape[0]):\n",
      "                    # This returns a series indexed by taxID.\n",
      "                    patient_16s_data_bodysite_dupEntry=patient_16s_data_bodysite.ix[i]\n",
      "                    patient_16s_data_bodysite_dupEntry.drop('',inplace=True)\n",
      "                    patient_16s_data_bodysite_dupEntry.drop('tax_id',inplace=True)\n",
      "                    patient_16s_data_bodysite_dupEntry=patient_16s_data_bodysite_dupEntry.astype(float)\n",
      "                    # Aggregate data for multiple instances of a given taxID.\n",
      "                    patient_16s_data_bodysite_grouped=patient_16s_data_bodysite_dupEntry.groupby(patient_16s_data_bodysite_dupEntry.index).sum()\n",
      "                    patient_16s_toConcat=pd.DataFrame(patient_16s_data_bodysite_grouped)\n",
      "                    tmp_stor=tmp_stor+[patient_16s_toConcat]\n",
      "                tmp_concat=pd.concat(tmp_stor,axis=1)\n",
      "\n",
      "                # Take the mean from duplicate body site measurments - \n",
      "                tmp_concat_toConcat=pd.DataFrame(tmp_concat.mean(axis=1))\n",
      "                tmp_concat_toConcat.columns=[bodysite]\n",
      "                storeData=storeData+[tmp_concat_toConcat]\n",
      "\n",
      "            # If there is a single body site measurment, Indexerror above is thrown. Process the series in isolation.\n",
      "            except IndexError:\n",
      "                patient_16s_data_bodysite.drop('',inplace=True)\n",
      "                patient_16s_data_bodysite.drop('tax_id',inplace=True)\n",
      "                patient_16s_data_bodysite=patient_16s_data_bodysite.astype(float)\n",
      "                patient_16s_data_bodysite_grouped=patient_16s_data_bodysite.groupby(patient_16s_data_bodysite.index).sum()\n",
      "                patient_16s_toConcat=pd.DataFrame(patient_16s_data_bodysite_grouped)\n",
      "                patient_16s_toConcat.columns=[bodysite]\n",
      "                storeData=storeData+[patient_16s_toConcat]\n",
      "    \n",
      "    if flag_16s and flag_cf:\n",
      "        # List of data from all body sites for samples. Add the corresponding cell-free DNA (blood) data. \n",
      "        storeData=storeData+[blood_data_toConcat]\n",
      "        data_all_cat=pd.concat(storeData,axis=1)\n",
      "        data_all_cat.fillna(0,inplace=True)\n",
      "        data_all_cat_norm=data_all_cat\n",
      "    \n",
      "    elif flag_16s:\n",
      "        data_all_cat=pd.concat(storeData,axis=1)\n",
      "        data_all_cat.fillna(0,inplace=True)\n",
      "        data_all_cat_norm=data_all_cat\n",
      "        \n",
      "    else:\n",
      "        data_all_cat=blood_data_toConcat\n",
      "        data_all_cat.fillna(0,inplace=True)\n",
      "        data_all_cat_norm=data_all_cat\n",
      "        \n",
      "    # Sort data by load \n",
      "    sums=data_all_cat_norm.sum(axis=1)\n",
      "    sums.sort(ascending=False)\n",
      "    data_all_cat_norm=data_all_cat_norm.loc[sums.index]\n",
      "        \n",
      "    # Specify final level of aggregation desired (class is nice beacuse it make viewing the data easier).\n",
      "    level_to_view_data='genus'\n",
      "    FullTaxTable=pd.read_table(path,dtype=str)[[\"tax_id\",\"tax_name\",level_to_view_data]]\n",
      "    FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "    merge_data=pd.merge(data_all_cat_norm,FullTaxTable,left_on=data_all_cat_norm.index,right_on=\"tax_id\")\n",
      "    merge_data.set_index(['tax_id','tax_name'],inplace=True)\n",
      "    data_abs=merge_data.groupby([level_to_view_data],as_index=False).aggregate(np.sum)\n",
      "    data_abs.set_index(level_to_view_data,inplace=True)\n",
      "    data_abs_annot=pd.merge(data_abs,FullTaxTable,left_on=data_abs.index,right_on=\"tax_id\")\n",
      "        \n",
      "    return data_abs_annot,level_to_view_data\n",
      "\n",
      "# This will store all cfDNA data with matched 16s data --- \n",
      "tmp_store=[]\n",
      "\n",
      "for sample in consolodated_samples_for_patients:\n",
      "    \n",
      "    # Cleanup for concatenation\n",
      "    dat_sample,level_to_view_data=merge_16s_with_cfDNA(sample)\n",
      "    try:\n",
      "        \n",
      "        # Prepare sample data\n",
      "        dat_sample.set_index(['tax_id','tax_name'],inplace=True)\n",
      "        del dat_sample[level_to_view_data]\n",
      "        dat_sample_norm=dat_sample.div(dat_sample.sum(axis=0),axis=1)\n",
      "        dat_sample_norm.fillna(0,inplace=True)\n",
      "        \n",
      "        # Add sample name to the index \n",
      "        dat_sample_tp=dat_sample.transpose()\n",
      "        dat_sample_tp['Sample']=sample\n",
      "        dat_sample_tp.set_index('Sample',append=True,inplace=True,drop=True)\n",
      "        dat_sample_withName=dat_sample_tp.transpose()\n",
      "        \n",
      "        # Store\n",
      "        tmp_store=tmp_store+[dat_sample_withName]\n",
      "        \n",
      "    except AttributeError:\n",
      "        None \n",
      "        # print \"No 16s data for %s\"%sample\n",
      "        \n",
      "# Concatenate data - \n",
      "data_all_16s_samples_bodysites=pd.concat(tmp_store,axis=1)\n",
      "data_all_16s_samples_bodysites.fillna(0,inplace=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "-c:35: SettingWithCopyWarning: A value is trying to be set on a copy of a slice from a DataFrame.\n",
        "Try using .loc[row_index,col_indexer] = value instead\n",
        "-c:115: SettingWithCopyWarning: A value is trying to be set on a copy of a slice from a DataFrame\n"
       ]
      }
     ],
     "prompt_number": 74
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "data_all_16s_samples_bodysites.columns.names = ['BodySite','Sample']\n",
      "outfilepathToSave=os.getcwd()+'/PR_sample_data_all16s_%s.csv'%taxLevel_taxTable\n",
      "data_all_16s_samples_bodysites.to_csv(outfilepathToSave)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 79
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Process HMP data - "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def makeSampleDict(pathToDict):\n",
      "    d={}\n",
      "    with open(pathToDict, 'r') as infile:\n",
      "        for line in infile:\n",
      "            # First field (comma delimited) is the name\t\n",
      "            sampleName=line.rstrip().split(',')[0].strip('\"')\n",
      "            # Second field (comma delimited) is the body site\t\n",
      "            bodySite=line.rstrip().split(',')[1].strip('\"')\n",
      "            d[sampleName]=bodySite\n",
      "    return d\n",
      "\n",
      "def getBodySites(path):\n",
      "    # Dictionary mapping HMP sample name to body site\n",
      "    pathToDict=path+\"/sample_bodySite\"\n",
      "    sampleToBodySite=makeSampleDict(pathToDict)\n",
      "    # Dictionary mapping HMP sample name to body site\n",
      "    sampleNames=path+\"/HMP_sample_names\"\n",
      "    samplesNames=np.genfromtxt(sampleNames,dtype=str,delimiter='\\t')\n",
      "    bodySites=[]\n",
      "    for name in samplesNames:\n",
      "        try:\n",
      "            site=sampleToBodySite[name]\n",
      "            bodySites=bodySites+[site]\n",
      "        except KeyError:\n",
      "            flag=0\n",
      "    return  bodySites\n",
      "\n",
      "# Column body site labels.\n",
      "path=os.getcwd()+'/reference/'\n",
      "HMPdataBodySites=list(getBodySites(path))\n",
      "\n",
      "# Column of microbes and matrix of abundances.\n",
      "hmpData=path+\"HMP_rawdata\"\n",
      "hmpDF=pd.DataFrame(pd.read_table(hmpData,header=None))\n",
      "hmpDF=hmpDF.set_index([0]) # Rows are bugs (718 total)\n",
      "hmpDF.columns=HMPdataBodySites # Columns are body sites for each sample (690)\n",
      "hmpDF.index.name='Bug'\n",
      "hmpDF.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>stool</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>...</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>stool</th>\n",
        "      <th>posterior_fornix</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Bug</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae</th>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.8352</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae</th>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.0000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae</th>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.0000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 690 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 37,
       "text": [
        "                                                                                              stool  \\\n",
        "Bug                                                                                                   \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae          0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae      0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae          0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                              tongue_dorsum  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                              tongue_dorsum  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                              tongue_dorsum  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              buccal_mucosa  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                                  ...        \\\n",
        "Bug                                                                                               ...         \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae          ...         \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae      ...         \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae          ...         \n",
        "\n",
        "                                                                                              buccal_mucosa  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              vaginal_introitus  \\\n",
        "Bug                                                                                                               \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                      0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                      0   \n",
        "\n",
        "                                                                                              mid_vagina  \\\n",
        "Bug                                                                                                        \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae               0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae           0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae               0   \n",
        "\n",
        "                                                                                              tongue_dorsum  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              anterior_nares  \\\n",
        "Bug                                                                                                            \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                   0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae               0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                   0   \n",
        "\n",
        "                                                                                              buccal_mucosa  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              tongue_dorsum  \\\n",
        "Bug                                                                                                           \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                  0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae              0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                  0   \n",
        "\n",
        "                                                                                              supragingival_plaque  \\\n",
        "Bug                                                                                                                  \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                         0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                     0   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                         0   \n",
        "\n",
        "                                                                                               stool  \\\n",
        "Bug                                                                                                    \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae      0.8352   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae  0.0000   \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae      0.0000   \n",
        "\n",
        "                                                                                              posterior_fornix  \n",
        "Bug                                                                                                             \n",
        "k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae                     0  \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhodospirillales|f__Acetobacteraceae                 0  \n",
        "k__Bacteria|p__Proteobacteria|c__Alphaproteobacteria|o__Rhizobiales|f__Bradyrhizobiaceae                     0  \n",
        "\n",
        "[3 rows x 690 columns]"
       ]
      }
     ],
     "prompt_number": 37
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "check=[bug for bug in hmpDF.index if 'Cup' in bug]\n",
      "check"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 113,
       "text": [
        "[]"
       ]
      }
     ],
     "prompt_number": 113
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Slice out genus\n",
      "tax_level_to_extract='g'\n",
      "tax_flags=np.array([tax.split('|')[-1].split('__')[0] for tax in hmpDF.index])\n",
      "all_genus=hmpDF.loc[tax_flags==tax_level_to_extract]\n",
      "names=np.array([tax.split('|')[-1].split('__')[1] for tax in all_genus.index])\n",
      "all_genus.index=names\n",
      "all_genus.index.name='Genus_ID'\n",
      "all_genus.columns.name='Bodysite'\n",
      "all_genus.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th>Bodysite</th>\n",
        "      <th>stool</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>...</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>stool</th>\n",
        "      <th>posterior_fornix</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Genus_ID</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Ruminococcus</th>\n",
        "      <td> 0.09916</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00011</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00117</td>\n",
        "      <td> 12.55463</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Agrobacterium</th>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.50109</td>\n",
        "      <td> 0.18917</td>\n",
        "      <td> 0.19244</td>\n",
        "      <td> 0.52411</td>\n",
        "      <td> 0.00753</td>\n",
        "      <td> 1.89595</td>\n",
        "      <td> 1.83749</td>\n",
        "      <td> 0.07781</td>\n",
        "      <td> 0.00923</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.82717</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.10741</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.18624</td>\n",
        "      <td> 0.10996</td>\n",
        "      <td> 1.97287</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 690 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 52,
       "text": [
        "Bodysite           stool  supragingival_plaque  supragingival_plaque  \\\n",
        "Genus_ID                                                               \n",
        "Ruminococcus     0.09916               0.00000               0.00000   \n",
        "Agrobacterium    0.00000               0.00000               0.00000   \n",
        "Aggregatibacter  0.00000               0.50109               0.18917   \n",
        "\n",
        "Bodysite         tongue_dorsum  supragingival_plaque  tongue_dorsum  \\\n",
        "Genus_ID                                                              \n",
        "Ruminococcus           0.00000               0.00000        0.00000   \n",
        "Agrobacterium          0.00000               0.00000        0.00000   \n",
        "Aggregatibacter        0.19244               0.52411        0.00753   \n",
        "\n",
        "Bodysite         supragingival_plaque  supragingival_plaque  tongue_dorsum  \\\n",
        "Genus_ID                                                                     \n",
        "Ruminococcus                  0.00000               0.00000        0.00011   \n",
        "Agrobacterium                 0.00000               0.00000        0.00000   \n",
        "Aggregatibacter               1.89595               1.83749        0.07781   \n",
        "\n",
        "Bodysite         buccal_mucosa      ...        buccal_mucosa  \\\n",
        "Genus_ID                            ...                        \n",
        "Ruminococcus           0.00000      ...              0.00000   \n",
        "Agrobacterium          0.00000      ...              0.00000   \n",
        "Aggregatibacter        0.00923      ...              0.82717   \n",
        "\n",
        "Bodysite         vaginal_introitus  mid_vagina  tongue_dorsum  anterior_nares  \\\n",
        "Genus_ID                                                                        \n",
        "Ruminococcus                     0           0        0.00000               0   \n",
        "Agrobacterium                    0           0        0.00000               0   \n",
        "Aggregatibacter                  0           0        0.10741               0   \n",
        "\n",
        "Bodysite         buccal_mucosa  tongue_dorsum  supragingival_plaque     stool  \\\n",
        "Genus_ID                                                                        \n",
        "Ruminococcus           0.00000        0.00000               0.00117  12.55463   \n",
        "Agrobacterium          0.00000        0.00000               0.00000   0.00000   \n",
        "Aggregatibacter        0.18624        0.10996               1.97287   0.00000   \n",
        "\n",
        "Bodysite         posterior_fornix  \n",
        "Genus_ID                           \n",
        "Ruminococcus                    0  \n",
        "Agrobacterium                   0  \n",
        "Aggregatibacter                 0  \n",
        "\n",
        "[3 rows x 690 columns]"
       ]
      }
     ],
     "prompt_number": 52
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "all_genus_tp=all_genus.transpose()\n",
      "all_genus_tp_mean=all_genus_tp.groupby(all_genus_tp.index).apply(lambda x:np.mean(x,axis=0))\n",
      "all_genus_mean=all_genus_tp_mean.transpose()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 59
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "all_genus_mean['tax_name']=all_genus_mean.index\n",
      "all_genus_mean.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th>Bodysite</th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>left_retroauricular_crease</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>palatine_tonsils</th>\n",
        "      <th>posterior_fornix</th>\n",
        "      <th>right_retroauricular_crease</th>\n",
        "      <th>saliva</th>\n",
        "      <th>stool</th>\n",
        "      <th>subgingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>throat</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "      <th>tax_name</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Genus_ID</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Ruminococcus</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007111</td>\n",
        "      <td> 0.021940</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.006873</td>\n",
        "      <td> 0.003139</td>\n",
        "      <td> 0.000914</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 3.701499</td>\n",
        "      <td> 0.000230</td>\n",
        "      <td> 0.001331</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00041</td>\n",
        "      <td> 0</td>\n",
        "      <td>    Ruminococcus</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Agrobacterium</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000008</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "      <td>   Agrobacterium</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <td> 0.001634</td>\n",
        "      <td> 0.079942</td>\n",
        "      <td> 0.217416</td>\n",
        "      <td> 0.015893</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.050678</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000434</td>\n",
        "      <td> 2.20616</td>\n",
        "      <td> 0.000107</td>\n",
        "      <td> 2.010644</td>\n",
        "      <td> 2.338830</td>\n",
        "      <td> 0.030823</td>\n",
        "      <td> 0.06349</td>\n",
        "      <td> 0</td>\n",
        "      <td> Aggregatibacter</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 68,
       "text": [
        "Bodysite         anterior_nares  attached_keratinized_gingiva  buccal_mucosa  \\\n",
        "Genus_ID                                                                       \n",
        "Ruminococcus           0.036996                      0.000000       0.007111   \n",
        "Agrobacterium          0.000000                      0.000000       0.000000   \n",
        "Aggregatibacter        0.001634                      0.079942       0.217416   \n",
        "\n",
        "Bodysite         left_retroauricular_crease  mid_vagina  palatine_tonsils  \\\n",
        "Genus_ID                                                                    \n",
        "Ruminococcus                       0.021940           0          0.006873   \n",
        "Agrobacterium                      0.000000           0          0.000000   \n",
        "Aggregatibacter                    0.015893           0          0.050678   \n",
        "\n",
        "Bodysite         posterior_fornix  right_retroauricular_crease   saliva  \\\n",
        "Genus_ID                                                                  \n",
        "Ruminococcus             0.003139                     0.000914  0.00000   \n",
        "Agrobacterium            0.000000                     0.000008  0.00000   \n",
        "Aggregatibacter          0.000000                     0.000434  2.20616   \n",
        "\n",
        "Bodysite            stool  subgingival_plaque  supragingival_plaque    throat  \\\n",
        "Genus_ID                                                                        \n",
        "Ruminococcus     3.701499            0.000230              0.001331  0.000000   \n",
        "Agrobacterium    0.000000            0.000000              0.000000  0.000000   \n",
        "Aggregatibacter  0.000107            2.010644              2.338830  0.030823   \n",
        "\n",
        "Bodysite         tongue_dorsum  vaginal_introitus         tax_name  \n",
        "Genus_ID                                                            \n",
        "Ruminococcus           0.00041                  0     Ruminococcus  \n",
        "Agrobacterium          0.00000                  0    Agrobacterium  \n",
        "Aggregatibacter        0.06349                  0  Aggregatibacter  "
       ]
      }
     ],
     "prompt_number": 68
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "path=\"/quakestor2/kertesz/soup/Data/NCBI/tax/Taxtastic/min.tax.table\"\n",
      "FullTaxTable=pd.read_table(path,dtype=str)[[\"tax_id\",\"tax_name\"]]\n",
      "FullTaxTable.set_index('tax_id',inplace=True,drop=False)\n",
      "FullTaxTable.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "      <th>tax_name</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_id</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
        "      <td>      1</td>\n",
        "      <td>               root</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>131567</th>\n",
        "      <td> 131567</td>\n",
        "      <td> cellular organisms</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2157</th>\n",
        "      <td>   2157</td>\n",
        "      <td>            Archaea</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 75,
       "text": [
        "        tax_id            tax_name\n",
        "tax_id                            \n",
        "1            1                root\n",
        "131567  131567  cellular organisms\n",
        "2157      2157             Archaea"
       ]
      }
     ],
     "prompt_number": 75
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm=pd.merge(all_genus_mean,FullTaxTable,left_on=\"tax_name\",right_on=\"tax_name\")\n",
      "mm.set_index(['tax_name','tax_id'],inplace=True)\n",
      "mm.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>left_retroauricular_crease</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>palatine_tonsils</th>\n",
        "      <th>posterior_fornix</th>\n",
        "      <th>right_retroauricular_crease</th>\n",
        "      <th>saliva</th>\n",
        "      <th>stool</th>\n",
        "      <th>subgingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>throat</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_name</th>\n",
        "      <th>tax_id</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Ruminococcus</th>\n",
        "      <th>1263</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007111</td>\n",
        "      <td> 0.021940</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.006873</td>\n",
        "      <td> 0.003139</td>\n",
        "      <td> 0.000914</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 3.701499</td>\n",
        "      <td> 0.000230</td>\n",
        "      <td> 0.001331</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00041</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Agrobacterium</th>\n",
        "      <th>357</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000008</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <th>416916</th>\n",
        "      <td> 0.001634</td>\n",
        "      <td> 0.079942</td>\n",
        "      <td> 0.217416</td>\n",
        "      <td> 0.015893</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.050678</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000434</td>\n",
        "      <td> 2.20616</td>\n",
        "      <td> 0.000107</td>\n",
        "      <td> 2.010644</td>\n",
        "      <td> 2.338830</td>\n",
        "      <td> 0.030823</td>\n",
        "      <td> 0.06349</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 82,
       "text": [
        "                        anterior_nares  attached_keratinized_gingiva  \\\n",
        "tax_name        tax_id                                                 \n",
        "Ruminococcus    1263          0.036996                      0.000000   \n",
        "Agrobacterium   357           0.000000                      0.000000   \n",
        "Aggregatibacter 416916        0.001634                      0.079942   \n",
        "\n",
        "                        buccal_mucosa  left_retroauricular_crease  mid_vagina  \\\n",
        "tax_name        tax_id                                                          \n",
        "Ruminococcus    1263         0.007111                    0.021940           0   \n",
        "Agrobacterium   357          0.000000                    0.000000           0   \n",
        "Aggregatibacter 416916       0.217416                    0.015893           0   \n",
        "\n",
        "                        palatine_tonsils  posterior_fornix  \\\n",
        "tax_name        tax_id                                       \n",
        "Ruminococcus    1263            0.006873          0.003139   \n",
        "Agrobacterium   357             0.000000          0.000000   \n",
        "Aggregatibacter 416916          0.050678          0.000000   \n",
        "\n",
        "                        right_retroauricular_crease   saliva     stool  \\\n",
        "tax_name        tax_id                                                   \n",
        "Ruminococcus    1263                       0.000914  0.00000  3.701499   \n",
        "Agrobacterium   357                        0.000008  0.00000  0.000000   \n",
        "Aggregatibacter 416916                     0.000434  2.20616  0.000107   \n",
        "\n",
        "                        subgingival_plaque  supragingival_plaque    throat  \\\n",
        "tax_name        tax_id                                                       \n",
        "Ruminococcus    1263              0.000230              0.001331  0.000000   \n",
        "Agrobacterium   357               0.000000              0.000000  0.000000   \n",
        "Aggregatibacter 416916            2.010644              2.338830  0.030823   \n",
        "\n",
        "                        tongue_dorsum  vaginal_introitus  \n",
        "tax_name        tax_id                                    \n",
        "Ruminococcus    1263          0.00041                  0  \n",
        "Agrobacterium   357           0.00000                  0  \n",
        "Aggregatibacter 416916        0.06349                  0  "
       ]
      }
     ],
     "prompt_number": 82
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm.columns.name='specific_site'\n",
      "mm_tp=mm.transpose()\n",
      "mm_tp.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr>\n",
        "      <th>tax_name</th>\n",
        "      <th>Ruminococcus</th>\n",
        "      <th>Agrobacterium</th>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <th>Olsenella</th>\n",
        "      <th>Citrobacter</th>\n",
        "      <th>Akkermansia</th>\n",
        "      <th>Leptotrichia</th>\n",
        "      <th>Jonquetella</th>\n",
        "      <th>Shuttleworthia</th>\n",
        "      <th>Psychrobacter</th>\n",
        "      <th>...</th>\n",
        "      <th>Actinomyces</th>\n",
        "      <th>Roseburia</th>\n",
        "      <th>Acidovorax</th>\n",
        "      <th>Rhodobacter</th>\n",
        "      <th>Treponema</th>\n",
        "      <th>Rhodopseudomonas</th>\n",
        "      <th>Porphyromonas</th>\n",
        "      <th>Parascardovia</th>\n",
        "      <th>Sutterella</th>\n",
        "      <th>Parvimonas</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_id</th>\n",
        "      <th>1263</th>\n",
        "      <th>357</th>\n",
        "      <th>416916</th>\n",
        "      <th>133925</th>\n",
        "      <th>544</th>\n",
        "      <th>239934</th>\n",
        "      <th>32067</th>\n",
        "      <th>428711</th>\n",
        "      <th>177971</th>\n",
        "      <th>497</th>\n",
        "      <th>...</th>\n",
        "      <th>1654</th>\n",
        "      <th>841</th>\n",
        "      <th>12916</th>\n",
        "      <th>1060</th>\n",
        "      <th>157</th>\n",
        "      <th>1073</th>\n",
        "      <th>836</th>\n",
        "      <th>196082</th>\n",
        "      <th>40544</th>\n",
        "      <th>543311</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>specific_site</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>anterior_nares</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.001634</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.002579</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.015932</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000199</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.079942</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.134162</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 1.050380</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000543</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.005852</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.039357</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007138</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <td> 0.007111</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.217416</td>\n",
        "      <td> 0.000557</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000504</td>\n",
        "      <td> 0.338077</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001011</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 1.401375</td>\n",
        "      <td> 0.002705</td>\n",
        "      <td> 0.006534</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.021058</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.077865</td>\n",
        "      <td> 0.000416</td>\n",
        "      <td> 0.000192</td>\n",
        "      <td> 0.028213</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 139 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 103,
       "text": [
        "tax_name                      Ruminococcus  Agrobacterium  Aggregatibacter  \\\n",
        "tax_id                                1263            357           416916   \n",
        "specific_site                                                                \n",
        "anterior_nares                    0.036996              0         0.001634   \n",
        "attached_keratinized_gingiva      0.000000              0         0.079942   \n",
        "buccal_mucosa                     0.007111              0         0.217416   \n",
        "\n",
        "tax_name                      Olsenella  Citrobacter  Akkermansia  \\\n",
        "tax_id                           133925          544       239934   \n",
        "specific_site                                                       \n",
        "anterior_nares                 0.000000            0     0.000000   \n",
        "attached_keratinized_gingiva   0.000000            0     0.000000   \n",
        "buccal_mucosa                  0.000557            0     0.000504   \n",
        "\n",
        "tax_name                      Leptotrichia  Jonquetella  Shuttleworthia  \\\n",
        "tax_id                               32067       428711          177971   \n",
        "specific_site                                                             \n",
        "anterior_nares                    0.000000     0.002579        0.000000   \n",
        "attached_keratinized_gingiva      0.134162     0.000000        0.000000   \n",
        "buccal_mucosa                     0.338077     0.000000        0.001011   \n",
        "\n",
        "tax_name                      Psychrobacter      ...        Actinomyces  \\\n",
        "tax_id                                  497      ...               1654   \n",
        "specific_site                                    ...                      \n",
        "anterior_nares                            0      ...           0.015932   \n",
        "attached_keratinized_gingiva              0      ...           1.050380   \n",
        "buccal_mucosa                             0      ...           1.401375   \n",
        "\n",
        "tax_name                      Roseburia  Acidovorax  Rhodobacter  Treponema  \\\n",
        "tax_id                              841       12916         1060        157   \n",
        "specific_site                                                                 \n",
        "anterior_nares                 0.000000    0.000199            0   0.000000   \n",
        "attached_keratinized_gingiva   0.000000    0.000543            0   0.005852   \n",
        "buccal_mucosa                  0.002705    0.006534            0   0.021058   \n",
        "\n",
        "tax_name                      Rhodopseudomonas  Porphyromonas  Parascardovia  \\\n",
        "tax_id                                    1073            836         196082   \n",
        "specific_site                                                                  \n",
        "anterior_nares                               0       0.000000       0.000000   \n",
        "attached_keratinized_gingiva                 0       0.039357       0.000000   \n",
        "buccal_mucosa                                0       0.077865       0.000416   \n",
        "\n",
        "tax_name                      Sutterella  Parvimonas  \n",
        "tax_id                             40544      543311  \n",
        "specific_site                                         \n",
        "anterior_nares                  0.000000    0.000000  \n",
        "attached_keratinized_gingiva    0.000000    0.007138  \n",
        "buccal_mucosa                   0.000192    0.028213  \n",
        "\n",
        "[3 rows x 139 columns]"
       ]
      }
     ],
     "prompt_number": 103
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "general_name=pd.Series(['skin','oral','oral','skin','vagina','oral','vagina','skin','oral','stool','oral','oral','throat','throat','vagina'])\n",
      "mm_tp.set_index(general_name,append=True,inplace=True)\n",
      "mm_tp.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>tax_name</th>\n",
        "      <th>Ruminococcus</th>\n",
        "      <th>Agrobacterium</th>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <th>Olsenella</th>\n",
        "      <th>Citrobacter</th>\n",
        "      <th>Akkermansia</th>\n",
        "      <th>Leptotrichia</th>\n",
        "      <th>Jonquetella</th>\n",
        "      <th>Shuttleworthia</th>\n",
        "      <th>Psychrobacter</th>\n",
        "      <th>...</th>\n",
        "      <th>Actinomyces</th>\n",
        "      <th>Roseburia</th>\n",
        "      <th>Acidovorax</th>\n",
        "      <th>Rhodobacter</th>\n",
        "      <th>Treponema</th>\n",
        "      <th>Rhodopseudomonas</th>\n",
        "      <th>Porphyromonas</th>\n",
        "      <th>Parascardovia</th>\n",
        "      <th>Sutterella</th>\n",
        "      <th>Parvimonas</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>tax_id</th>\n",
        "      <th>1263</th>\n",
        "      <th>357</th>\n",
        "      <th>416916</th>\n",
        "      <th>133925</th>\n",
        "      <th>544</th>\n",
        "      <th>239934</th>\n",
        "      <th>32067</th>\n",
        "      <th>428711</th>\n",
        "      <th>177971</th>\n",
        "      <th>497</th>\n",
        "      <th>...</th>\n",
        "      <th>1654</th>\n",
        "      <th>841</th>\n",
        "      <th>12916</th>\n",
        "      <th>1060</th>\n",
        "      <th>157</th>\n",
        "      <th>1073</th>\n",
        "      <th>836</th>\n",
        "      <th>196082</th>\n",
        "      <th>40544</th>\n",
        "      <th>543311</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>specific_site</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>skin</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.001634</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.002579</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.015932</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000199</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <th>oral</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.079942</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.134162</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 1.050380</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000543</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.005852</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.039357</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007138</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>oral</th>\n",
        "      <td> 0.007111</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.217416</td>\n",
        "      <td> 0.000557</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000504</td>\n",
        "      <td> 0.338077</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001011</td>\n",
        "      <td> 0</td>\n",
        "      <td>...</td>\n",
        "      <td> 1.401375</td>\n",
        "      <td> 0.002705</td>\n",
        "      <td> 0.006534</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.021058</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.077865</td>\n",
        "      <td> 0.000416</td>\n",
        "      <td> 0.000192</td>\n",
        "      <td> 0.028213</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>3 rows \u00d7 139 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 104,
       "text": [
        "tax_name                           Ruminococcus  Agrobacterium  \\\n",
        "tax_id                                     1263            357   \n",
        "specific_site                                                    \n",
        "anterior_nares               skin      0.036996              0   \n",
        "attached_keratinized_gingiva oral      0.000000              0   \n",
        "buccal_mucosa                oral      0.007111              0   \n",
        "\n",
        "tax_name                           Aggregatibacter  Olsenella  Citrobacter  \\\n",
        "tax_id                                      416916     133925          544   \n",
        "specific_site                                                                \n",
        "anterior_nares               skin         0.001634   0.000000            0   \n",
        "attached_keratinized_gingiva oral         0.079942   0.000000            0   \n",
        "buccal_mucosa                oral         0.217416   0.000557            0   \n",
        "\n",
        "tax_name                           Akkermansia  Leptotrichia  Jonquetella  \\\n",
        "tax_id                                  239934         32067       428711   \n",
        "specific_site                                                               \n",
        "anterior_nares               skin     0.000000      0.000000     0.002579   \n",
        "attached_keratinized_gingiva oral     0.000000      0.134162     0.000000   \n",
        "buccal_mucosa                oral     0.000504      0.338077     0.000000   \n",
        "\n",
        "tax_name                           Shuttleworthia  Psychrobacter  \\\n",
        "tax_id                                     177971            497   \n",
        "specific_site                                                      \n",
        "anterior_nares               skin        0.000000              0   \n",
        "attached_keratinized_gingiva oral        0.000000              0   \n",
        "buccal_mucosa                oral        0.001011              0   \n",
        "\n",
        "tax_name                               ...        Actinomyces  Roseburia  \\\n",
        "tax_id                                 ...               1654        841   \n",
        "specific_site                          ...                                 \n",
        "anterior_nares               skin      ...           0.015932   0.000000   \n",
        "attached_keratinized_gingiva oral      ...           1.050380   0.000000   \n",
        "buccal_mucosa                oral      ...           1.401375   0.002705   \n",
        "\n",
        "tax_name                           Acidovorax  Rhodobacter  Treponema  \\\n",
        "tax_id                                  12916         1060        157   \n",
        "specific_site                                                           \n",
        "anterior_nares               skin    0.000199            0   0.000000   \n",
        "attached_keratinized_gingiva oral    0.000543            0   0.005852   \n",
        "buccal_mucosa                oral    0.006534            0   0.021058   \n",
        "\n",
        "tax_name                           Rhodopseudomonas  Porphyromonas  \\\n",
        "tax_id                                         1073            836   \n",
        "specific_site                                                        \n",
        "anterior_nares               skin                 0       0.000000   \n",
        "attached_keratinized_gingiva oral                 0       0.039357   \n",
        "buccal_mucosa                oral                 0       0.077865   \n",
        "\n",
        "tax_name                           Parascardovia  Sutterella  Parvimonas  \n",
        "tax_id                                    196082       40544      543311  \n",
        "specific_site                                                             \n",
        "anterior_nares               skin       0.000000    0.000000    0.000000  \n",
        "attached_keratinized_gingiva oral       0.000000    0.000000    0.007138  \n",
        "buccal_mucosa                oral       0.000416    0.000192    0.028213  \n",
        "\n",
        "[3 rows x 139 columns]"
       ]
      }
     ],
     "prompt_number": 104
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm_tp.index.set_names(['specific_site','general_site'],inplace=True)\n",
      "mm=mm_tp.transpose()\n",
      "mm.head(3)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>specific_site</th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>left_retroauricular_crease</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>palatine_tonsils</th>\n",
        "      <th>posterior_fornix</th>\n",
        "      <th>right_retroauricular_crease</th>\n",
        "      <th>saliva</th>\n",
        "      <th>stool</th>\n",
        "      <th>subgingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>throat</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>general_site</th>\n",
        "      <th>skin</th>\n",
        "      <th>oral</th>\n",
        "      <th>oral</th>\n",
        "      <th>skin</th>\n",
        "      <th>vagina</th>\n",
        "      <th>oral</th>\n",
        "      <th>vagina</th>\n",
        "      <th>skin</th>\n",
        "      <th>oral</th>\n",
        "      <th>stool</th>\n",
        "      <th>oral</th>\n",
        "      <th>oral</th>\n",
        "      <th>throat</th>\n",
        "      <th>throat</th>\n",
        "      <th>vagina</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_name</th>\n",
        "      <th>tax_id</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Ruminococcus</th>\n",
        "      <th>1263</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007111</td>\n",
        "      <td> 0.021940</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.006873</td>\n",
        "      <td> 0.003139</td>\n",
        "      <td> 0.000914</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 3.701499</td>\n",
        "      <td> 0.000230</td>\n",
        "      <td> 0.001331</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00041</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Agrobacterium</th>\n",
        "      <th>357</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000008</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.00000</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <th>416916</th>\n",
        "      <td> 0.001634</td>\n",
        "      <td> 0.079942</td>\n",
        "      <td> 0.217416</td>\n",
        "      <td> 0.015893</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0.050678</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000434</td>\n",
        "      <td> 2.20616</td>\n",
        "      <td> 0.000107</td>\n",
        "      <td> 2.010644</td>\n",
        "      <td> 2.338830</td>\n",
        "      <td> 0.030823</td>\n",
        "      <td> 0.06349</td>\n",
        "      <td> 0</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 108,
       "text": [
        "specific_site           anterior_nares  attached_keratinized_gingiva  \\\n",
        "general_site                      skin                          oral   \n",
        "tax_name        tax_id                                                 \n",
        "Ruminococcus    1263          0.036996                      0.000000   \n",
        "Agrobacterium   357           0.000000                      0.000000   \n",
        "Aggregatibacter 416916        0.001634                      0.079942   \n",
        "\n",
        "specific_site           buccal_mucosa  left_retroauricular_crease  mid_vagina  \\\n",
        "general_site                     oral                        skin      vagina   \n",
        "tax_name        tax_id                                                          \n",
        "Ruminococcus    1263         0.007111                    0.021940           0   \n",
        "Agrobacterium   357          0.000000                    0.000000           0   \n",
        "Aggregatibacter 416916       0.217416                    0.015893           0   \n",
        "\n",
        "specific_site           palatine_tonsils  posterior_fornix  \\\n",
        "general_site                        oral            vagina   \n",
        "tax_name        tax_id                                       \n",
        "Ruminococcus    1263            0.006873          0.003139   \n",
        "Agrobacterium   357             0.000000          0.000000   \n",
        "Aggregatibacter 416916          0.050678          0.000000   \n",
        "\n",
        "specific_site           right_retroauricular_crease   saliva     stool  \\\n",
        "general_site                                   skin     oral     stool   \n",
        "tax_name        tax_id                                                   \n",
        "Ruminococcus    1263                       0.000914  0.00000  3.701499   \n",
        "Agrobacterium   357                        0.000008  0.00000  0.000000   \n",
        "Aggregatibacter 416916                     0.000434  2.20616  0.000107   \n",
        "\n",
        "specific_site           subgingival_plaque  supragingival_plaque    throat  \\\n",
        "general_site                          oral                  oral    throat   \n",
        "tax_name        tax_id                                                       \n",
        "Ruminococcus    1263              0.000230              0.001331  0.000000   \n",
        "Agrobacterium   357               0.000000              0.000000  0.000000   \n",
        "Aggregatibacter 416916            2.010644              2.338830  0.030823   \n",
        "\n",
        "specific_site           tongue_dorsum  vaginal_introitus  \n",
        "general_site                   throat             vagina  \n",
        "tax_name        tax_id                                    \n",
        "Ruminococcus    1263          0.00041                  0  \n",
        "Agrobacterium   357           0.00000                  0  \n",
        "Aggregatibacter 416916        0.06349                  0  "
       ]
      }
     ],
     "prompt_number": 108
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "outfilepathToSave=os.getcwd()+'/HMP_data_%s.csv'%'genus'\n",
      "mm.to_csv(outfilepathToSave)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 109
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>specific_site</th>\n",
        "      <th>anterior_nares</th>\n",
        "      <th>attached_keratinized_gingiva</th>\n",
        "      <th>buccal_mucosa</th>\n",
        "      <th>left_retroauricular_crease</th>\n",
        "      <th>mid_vagina</th>\n",
        "      <th>palatine_tonsils</th>\n",
        "      <th>posterior_fornix</th>\n",
        "      <th>right_retroauricular_crease</th>\n",
        "      <th>saliva</th>\n",
        "      <th>stool</th>\n",
        "      <th>subgingival_plaque</th>\n",
        "      <th>supragingival_plaque</th>\n",
        "      <th>throat</th>\n",
        "      <th>tongue_dorsum</th>\n",
        "      <th>vaginal_introitus</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <th>general_site</th>\n",
        "      <th>skin</th>\n",
        "      <th>oral</th>\n",
        "      <th>oral</th>\n",
        "      <th>skin</th>\n",
        "      <th>vagina</th>\n",
        "      <th>oral</th>\n",
        "      <th>vagina</th>\n",
        "      <th>skin</th>\n",
        "      <th>oral</th>\n",
        "      <th>stool</th>\n",
        "      <th>oral</th>\n",
        "      <th>oral</th>\n",
        "      <th>throat</th>\n",
        "      <th>throat</th>\n",
        "      <th>vagina</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tax_name</th>\n",
        "      <th>tax_id</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Ruminococcus</th>\n",
        "      <th>1263</th>\n",
        "      <td> 0.036996</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.007111</td>\n",
        "      <td> 0.021940</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.006873</td>\n",
        "      <td>  0.003139</td>\n",
        "      <td> 0.000914</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 3.701499</td>\n",
        "      <td> 0.000230</td>\n",
        "      <td> 0.001331</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000410</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Agrobacterium</th>\n",
        "      <th>357</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000008</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Aggregatibacter</th>\n",
        "      <th>416916</th>\n",
        "      <td> 0.001634</td>\n",
        "      <td>  0.079942</td>\n",
        "      <td>  0.217416</td>\n",
        "      <td> 0.015893</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.050678</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000434</td>\n",
        "      <td>  2.206160</td>\n",
        "      <td> 0.000107</td>\n",
        "      <td> 2.010644</td>\n",
        "      <td> 2.338830</td>\n",
        "      <td>  0.030823</td>\n",
        "      <td>  0.063490</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Olsenella</th>\n",
        "      <th>133925</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000557</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.101300</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.001173</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.056920</td>\n",
        "      <td> 0.002011</td>\n",
        "      <td>  0.008174</td>\n",
        "      <td>  0.000377</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Citrobacter</th>\n",
        "      <th>544</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000883</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Akkermansia</th>\n",
        "      <th>239934</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000504</td>\n",
        "      <td> 0.000749</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000034</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.899293</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000012</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000006</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Leptotrichia</th>\n",
        "      <th>32067</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.134162</td>\n",
        "      <td>  0.338077</td>\n",
        "      <td> 0.003350</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.518070</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.001141</td>\n",
        "      <td>  0.185980</td>\n",
        "      <td> 0.000024</td>\n",
        "      <td> 1.960361</td>\n",
        "      <td> 2.123331</td>\n",
        "      <td>  0.304966</td>\n",
        "      <td>  0.573608</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Jonquetella</th>\n",
        "      <th>428711</th>\n",
        "      <td> 0.002579</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Shuttleworthia</th>\n",
        "      <th>177971</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.001011</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.014257</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.029916</td>\n",
        "      <td> 0.002324</td>\n",
        "      <td>  0.000353</td>\n",
        "      <td>  0.000428</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Psychrobacter</th>\n",
        "      <th>497</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.031707</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000005</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Pediococcus</th>\n",
        "      <th>1253</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000030</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Phascolarctobacterium</th>\n",
        "      <th>33024</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.142882</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Streptobacillus</th>\n",
        "      <th>34104</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.007440</td>\n",
        "      <td>  0.000207</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000002</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000002</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Enterococcus</th>\n",
        "      <th>1350</th>\n",
        "      <td> 0.034706</td>\n",
        "      <td>  0.041725</td>\n",
        "      <td>  0.028056</td>\n",
        "      <td> 0.000426</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.034695</td>\n",
        "      <td>  0.000458</td>\n",
        "      <td> 0.000067</td>\n",
        "      <td>  0.041723</td>\n",
        "      <td> 0.007888</td>\n",
        "      <td> 0.004329</td>\n",
        "      <td> 0.007862</td>\n",
        "      <td>  0.014853</td>\n",
        "      <td>  0.035602</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Megamonas</th>\n",
        "      <th>158846</th>\n",
        "      <td> 0.001062</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000886</td>\n",
        "      <td> 0.000024</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.042272</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000519</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000140</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Lactobacillus</th>\n",
        "      <th>1578</th>\n",
        "      <td> 0.005129</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.006150</td>\n",
        "      <td> 0.005456</td>\n",
        "      <td> 98.55888</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 92.393338</td>\n",
        "      <td> 0.088401</td>\n",
        "      <td>  0.098157</td>\n",
        "      <td> 0.012162</td>\n",
        "      <td> 0.000913</td>\n",
        "      <td> 0.001525</td>\n",
        "      <td>  0.000457</td>\n",
        "      <td>  0.000708</td>\n",
        "      <td> 88.898777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sphingopyxis</th>\n",
        "      <th>165697</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000076</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Methylobacterium</th>\n",
        "      <th>407</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000892</td>\n",
        "      <td> 0.000071</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000219</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000209</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000114</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000030</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Brevundimonas</th>\n",
        "      <th>41275</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000165</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Delftia</th>\n",
        "      <th>80865</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000121</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Bilophila</th>\n",
        "      <th>35832</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000169</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.065139</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000005</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Catenibacterium</th>\n",
        "      <th>135858</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.001126</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.056663</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Acidaminococcus</th>\n",
        "      <th>904</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000042</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Cardiobacterium</th>\n",
        "      <th>2717</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.003278</td>\n",
        "      <td>  0.031258</td>\n",
        "      <td> 0.000184</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000337</td>\n",
        "      <td>  0.083893</td>\n",
        "      <td> 0.000001</td>\n",
        "      <td> 2.559864</td>\n",
        "      <td> 1.420594</td>\n",
        "      <td>  0.002666</td>\n",
        "      <td>  0.023617</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Solobacterium</th>\n",
        "      <th>123375</th>\n",
        "      <td> 0.000019</td>\n",
        "      <td>  0.004997</td>\n",
        "      <td>  0.031583</td>\n",
        "      <td> 0.000077</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  1.004143</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.292127</td>\n",
        "      <td> 0.001283</td>\n",
        "      <td> 0.103901</td>\n",
        "      <td> 0.030559</td>\n",
        "      <td>  0.770401</td>\n",
        "      <td>  0.862904</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Subdoligranulum</th>\n",
        "      <th>292632</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000008</td>\n",
        "      <td> 0.000008</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.015273</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Lautropia</th>\n",
        "      <th>47670</th>\n",
        "      <td> 0.011475</td>\n",
        "      <td>  0.077440</td>\n",
        "      <td>  0.379796</td>\n",
        "      <td> 0.009278</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.005002</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.001015</td>\n",
        "      <td>  0.387200</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.954963</td>\n",
        "      <td> 2.685566</td>\n",
        "      <td>  0.018849</td>\n",
        "      <td>  0.098337</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Brucella</th>\n",
        "      <th>234</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000014</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Catonella</th>\n",
        "      <th>43996</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.005243</td>\n",
        "      <td>  0.006500</td>\n",
        "      <td> 0.004297</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.069165</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.030553</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.330303</td>\n",
        "      <td> 0.192797</td>\n",
        "      <td>  0.030833</td>\n",
        "      <td>  0.023674</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Lactococcus</th>\n",
        "      <th>1357</th>\n",
        "      <td> 0.043511</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.004625</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.002952</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000726</td>\n",
        "      <td>  0.003113</td>\n",
        "      <td> 0.011050</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001020</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.001197</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Actinobacillus</th>\n",
        "      <th>713</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000952</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.001320</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.007360</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000056</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000104</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Leuconostoc</th>\n",
        "      <th>1243</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.005990</td>\n",
        "      <td> 0.000227</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000010</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Holdemania</th>\n",
        "      <th>61170</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000010</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.026074</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Atopobium</th>\n",
        "      <th>1380</th>\n",
        "      <td> 0.002336</td>\n",
        "      <td>  0.014587</td>\n",
        "      <td>  0.068410</td>\n",
        "      <td> 0.000107</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.770162</td>\n",
        "      <td>  0.916763</td>\n",
        "      <td> 0.003071</td>\n",
        "      <td>  0.970937</td>\n",
        "      <td> 0.000291</td>\n",
        "      <td> 0.261836</td>\n",
        "      <td> 0.062533</td>\n",
        "      <td>  0.429936</td>\n",
        "      <td>  0.687797</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Proteus</th>\n",
        "      <th>583</th>\n",
        "      <td> 0.009402</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Marinomonas</th>\n",
        "      <th>28253</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000043</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000078</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000029</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Collinsella</th>\n",
        "      <th>102106</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000217</td>\n",
        "      <td> 0.006098</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.206767</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Ureaplasma</th>\n",
        "      <th>2129</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.07112</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.268771</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000014</td>\n",
        "      <td>  1.389137</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Megasphaera</th>\n",
        "      <th>906</th>\n",
        "      <td> 0.002331</td>\n",
        "      <td>  0.005897</td>\n",
        "      <td>  0.054954</td>\n",
        "      <td> 0.000068</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  1.748103</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  1.591690</td>\n",
        "      <td> 0.003123</td>\n",
        "      <td> 0.052337</td>\n",
        "      <td> 0.008704</td>\n",
        "      <td>  1.944736</td>\n",
        "      <td>  1.173358</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Serratia</th>\n",
        "      <th>613</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.102950</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000002</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Xanthomonas</th>\n",
        "      <th>338</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000012</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Herbaspirillum</th>\n",
        "      <th>963</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000020</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.008177</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Pasteurella</th>\n",
        "      <th>745</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000036</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000109</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000016</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Micrococcus</th>\n",
        "      <th>1269</th>\n",
        "      <td> 0.002031</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.005821</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.002174</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Cellulophaga</th>\n",
        "      <th>104264</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.001965</td>\n",
        "      <td>  0.003667</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.002908</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.007527</td>\n",
        "      <td> 0.000077</td>\n",
        "      <td> 0.046329</td>\n",
        "      <td> 0.049064</td>\n",
        "      <td>  0.013480</td>\n",
        "      <td>  0.007234</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Haemophilus</th>\n",
        "      <th>724</th>\n",
        "      <td> 0.625698</td>\n",
        "      <td> 11.542440</td>\n",
        "      <td> 16.233547</td>\n",
        "      <td> 0.084362</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td> 13.383828</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.076659</td>\n",
        "      <td> 20.294517</td>\n",
        "      <td> 0.088559</td>\n",
        "      <td> 2.898726</td>\n",
        "      <td> 7.760013</td>\n",
        "      <td> 10.312489</td>\n",
        "      <td> 18.012251</td>\n",
        "      <td>  0.000413</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Oxalobacter</th>\n",
        "      <th>846</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.008908</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Alkaliphilus</th>\n",
        "      <th>114627</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000488</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.001738</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000587</td>\n",
        "      <td> 0.000444</td>\n",
        "      <td> 0.000229</td>\n",
        "      <td>  0.003694</td>\n",
        "      <td>  0.004875</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Faecalibacterium</th>\n",
        "      <th>216851</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.013195</td>\n",
        "      <td> 0.008172</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000082</td>\n",
        "      <td>  0.000802</td>\n",
        "      <td> 0.000154</td>\n",
        "      <td>  0.004523</td>\n",
        "      <td> 3.482948</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000017</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000092</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Escherichia</th>\n",
        "      <th>561</th>\n",
        "      <td> 0.051890</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.001387</td>\n",
        "      <td> 0.001069</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000460</td>\n",
        "      <td> 0.014660</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.809806</td>\n",
        "      <td> 0.321337</td>\n",
        "      <td> 0.000210</td>\n",
        "      <td>  0.000766</td>\n",
        "      <td>  0.000165</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Actinomyces</th>\n",
        "      <th>1654</th>\n",
        "      <td> 0.015932</td>\n",
        "      <td>  1.050380</td>\n",
        "      <td>  1.401375</td>\n",
        "      <td> 0.016103</td>\n",
        "      <td>  0.00377</td>\n",
        "      <td>  0.480800</td>\n",
        "      <td>  0.000415</td>\n",
        "      <td> 0.002789</td>\n",
        "      <td>  1.066360</td>\n",
        "      <td> 0.000181</td>\n",
        "      <td> 3.418323</td>\n",
        "      <td> 6.256130</td>\n",
        "      <td>  0.583140</td>\n",
        "      <td>  1.645602</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Roseburia</th>\n",
        "      <th>841</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.002705</td>\n",
        "      <td> 0.008171</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 1.451654</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000005</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Acidovorax</th>\n",
        "      <th>12916</th>\n",
        "      <td> 0.000199</td>\n",
        "      <td>  0.000543</td>\n",
        "      <td>  0.006534</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000235</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000262</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000022</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Rhodobacter</th>\n",
        "      <th>1060</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000023</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Treponema</th>\n",
        "      <th>157</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.005852</td>\n",
        "      <td>  0.021058</td>\n",
        "      <td> 0.000013</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.681318</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.067200</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 3.899533</td>\n",
        "      <td> 0.344508</td>\n",
        "      <td>  0.006640</td>\n",
        "      <td>  0.025036</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Rhodopseudomonas</th>\n",
        "      <th>1073</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000009</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Porphyromonas</th>\n",
        "      <th>836</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.039357</td>\n",
        "      <td>  0.077865</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.296672</td>\n",
        "      <td>  0.002486</td>\n",
        "      <td> 0.000682</td>\n",
        "      <td>  1.260640</td>\n",
        "      <td> 0.005276</td>\n",
        "      <td> 1.862417</td>\n",
        "      <td> 0.322816</td>\n",
        "      <td>  0.024234</td>\n",
        "      <td>  0.050447</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Parascardovia</th>\n",
        "      <th>196082</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000416</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000003</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sutterella</th>\n",
        "      <th>40544</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000192</td>\n",
        "      <td> 0.000004</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td> 1.099559</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000001</td>\n",
        "      <td>  0.000000</td>\n",
        "      <td>  0.000032</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Parvimonas</th>\n",
        "      <th>543311</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.007138</td>\n",
        "      <td>  0.028213</td>\n",
        "      <td> 0.002996</td>\n",
        "      <td>  0.00000</td>\n",
        "      <td>  1.553945</td>\n",
        "      <td>  0.000182</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>  0.208830</td>\n",
        "      <td> 0.001054</td>\n",
        "      <td> 2.574669</td>\n",
        "      <td> 0.126719</td>\n",
        "      <td>  0.305444</td>\n",
        "      <td>  0.149829</td>\n",
        "      <td>  0.000000</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>139 rows \u00d7 15 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 110,
       "text": [
        "specific_site                 anterior_nares  attached_keratinized_gingiva  \\\n",
        "general_site                            skin                          oral   \n",
        "tax_name              tax_id                                                 \n",
        "Ruminococcus          1263          0.036996                      0.000000   \n",
        "Agrobacterium         357           0.000000                      0.000000   \n",
        "Aggregatibacter       416916        0.001634                      0.079942   \n",
        "Olsenella             133925        0.000000                      0.000000   \n",
        "Citrobacter           544           0.000000                      0.000000   \n",
        "Akkermansia           239934        0.000000                      0.000000   \n",
        "Leptotrichia          32067         0.000000                      0.134162   \n",
        "Jonquetella           428711        0.002579                      0.000000   \n",
        "Shuttleworthia        177971        0.000000                      0.000000   \n",
        "Psychrobacter         497           0.000000                      0.000000   \n",
        "Pediococcus           1253          0.000000                      0.000000   \n",
        "Phascolarctobacterium 33024         0.000000                      0.000000   \n",
        "Streptobacillus       34104         0.000000                      0.007440   \n",
        "Enterococcus          1350          0.034706                      0.041725   \n",
        "Megamonas             158846        0.001062                      0.000000   \n",
        "Lactobacillus         1578          0.005129                      0.000000   \n",
        "Sphingopyxis          165697        0.000000                      0.000000   \n",
        "Methylobacterium      407           0.000000                      0.000000   \n",
        "Brevundimonas         41275         0.000000                      0.000000   \n",
        "Delftia               80865         0.000000                      0.000000   \n",
        "Bilophila             35832         0.000000                      0.000000   \n",
        "Catenibacterium       135858        0.000000                      0.000000   \n",
        "Acidaminococcus       904           0.000000                      0.000000   \n",
        "Cardiobacterium       2717          0.000000                      0.003278   \n",
        "Solobacterium         123375        0.000019                      0.004997   \n",
        "Subdoligranulum       292632        0.000000                      0.000000   \n",
        "Lautropia             47670         0.011475                      0.077440   \n",
        "Brucella              234           0.000000                      0.000000   \n",
        "Catonella             43996         0.000000                      0.005243   \n",
        "Lactococcus           1357          0.043511                      0.000000   \n",
        "...                                      ...                           ...   \n",
        "Actinobacillus        713           0.000000                      0.000000   \n",
        "Leuconostoc           1243          0.000000                      0.000000   \n",
        "Holdemania            61170         0.000000                      0.000000   \n",
        "Atopobium             1380          0.002336                      0.014587   \n",
        "Proteus               583           0.009402                      0.000000   \n",
        "Marinomonas           28253         0.000000                      0.000000   \n",
        "Collinsella           102106        0.000000                      0.000000   \n",
        "Ureaplasma            2129          0.000000                      0.000000   \n",
        "Megasphaera           906           0.002331                      0.005897   \n",
        "Serratia              613           0.000000                      0.000000   \n",
        "Xanthomonas           338           0.000000                      0.000000   \n",
        "Herbaspirillum        963           0.000000                      0.000000   \n",
        "Pasteurella           745           0.000000                      0.000000   \n",
        "Micrococcus           1269          0.002031                      0.000000   \n",
        "Cellulophaga          104264        0.000000                      0.001965   \n",
        "Haemophilus           724           0.625698                     11.542440   \n",
        "Oxalobacter           846           0.000000                      0.000000   \n",
        "Alkaliphilus          114627        0.000000                      0.000000   \n",
        "Faecalibacterium      216851        0.000000                      0.000000   \n",
        "Escherichia           561           0.051890                      0.000000   \n",
        "Actinomyces           1654          0.015932                      1.050380   \n",
        "Roseburia             841           0.000000                      0.000000   \n",
        "Acidovorax            12916         0.000199                      0.000543   \n",
        "Rhodobacter           1060          0.000000                      0.000000   \n",
        "Treponema             157           0.000000                      0.005852   \n",
        "Rhodopseudomonas      1073          0.000000                      0.000000   \n",
        "Porphyromonas         836           0.000000                      0.039357   \n",
        "Parascardovia         196082        0.000000                      0.000000   \n",
        "Sutterella            40544         0.000000                      0.000000   \n",
        "Parvimonas            543311        0.000000                      0.007138   \n",
        "\n",
        "specific_site                 buccal_mucosa  left_retroauricular_crease  \\\n",
        "general_site                           oral                        skin   \n",
        "tax_name              tax_id                                              \n",
        "Ruminococcus          1263         0.007111                    0.021940   \n",
        "Agrobacterium         357          0.000000                    0.000000   \n",
        "Aggregatibacter       416916       0.217416                    0.015893   \n",
        "Olsenella             133925       0.000557                    0.000000   \n",
        "Citrobacter           544          0.000000                    0.000000   \n",
        "Akkermansia           239934       0.000504                    0.000749   \n",
        "Leptotrichia          32067        0.338077                    0.003350   \n",
        "Jonquetella           428711       0.000000                    0.000000   \n",
        "Shuttleworthia        177971       0.001011                    0.000000   \n",
        "Psychrobacter         497          0.000000                    0.000000   \n",
        "Pediococcus           1253         0.000000                    0.000000   \n",
        "Phascolarctobacterium 33024        0.000000                    0.000000   \n",
        "Streptobacillus       34104        0.000207                    0.000000   \n",
        "Enterococcus          1350         0.028056                    0.000426   \n",
        "Megamonas             158846       0.000886                    0.000024   \n",
        "Lactobacillus         1578         0.006150                    0.005456   \n",
        "Sphingopyxis          165697       0.000000                    0.000000   \n",
        "Methylobacterium      407          0.000892                    0.000071   \n",
        "Brevundimonas         41275        0.000000                    0.000000   \n",
        "Delftia               80865        0.000000                    0.000000   \n",
        "Bilophila             35832        0.000169                    0.000000   \n",
        "Catenibacterium       135858       0.000000                    0.001126   \n",
        "Acidaminococcus       904          0.000000                    0.000000   \n",
        "Cardiobacterium       2717         0.031258                    0.000184   \n",
        "Solobacterium         123375       0.031583                    0.000077   \n",
        "Subdoligranulum       292632       0.000008                    0.000008   \n",
        "Lautropia             47670        0.379796                    0.009278   \n",
        "Brucella              234          0.000000                    0.000000   \n",
        "Catonella             43996        0.006500                    0.004297   \n",
        "Lactococcus           1357         0.004625                    0.000000   \n",
        "...                                     ...                         ...   \n",
        "Actinobacillus        713          0.000952                    0.000000   \n",
        "Leuconostoc           1243         0.000000                    0.000000   \n",
        "Holdemania            61170        0.000000                    0.000010   \n",
        "Atopobium             1380         0.068410                    0.000107   \n",
        "Proteus               583          0.000000                    0.000000   \n",
        "Marinomonas           28253        0.000043                    0.000000   \n",
        "Collinsella           102106       0.000217                    0.006098   \n",
        "Ureaplasma            2129         0.000000                    0.000000   \n",
        "Megasphaera           906          0.054954                    0.000068   \n",
        "Serratia              613          0.000000                    0.000000   \n",
        "Xanthomonas           338          0.000000                    0.000000   \n",
        "Herbaspirillum        963          0.000000                    0.000000   \n",
        "Pasteurella           745          0.000000                    0.000000   \n",
        "Micrococcus           1269         0.000000                    0.005821   \n",
        "Cellulophaga          104264       0.003667                    0.000000   \n",
        "Haemophilus           724         16.233547                    0.084362   \n",
        "Oxalobacter           846          0.000000                    0.000000   \n",
        "Alkaliphilus          114627       0.000488                    0.000000   \n",
        "Faecalibacterium      216851       0.013195                    0.008172   \n",
        "Escherichia           561          0.001387                    0.001069   \n",
        "Actinomyces           1654         1.401375                    0.016103   \n",
        "Roseburia             841          0.002705                    0.008171   \n",
        "Acidovorax            12916        0.006534                    0.000000   \n",
        "Rhodobacter           1060         0.000000                    0.000000   \n",
        "Treponema             157          0.021058                    0.000013   \n",
        "Rhodopseudomonas      1073         0.000000                    0.000000   \n",
        "Porphyromonas         836          0.077865                    0.000000   \n",
        "Parascardovia         196082       0.000416                    0.000000   \n",
        "Sutterella            40544        0.000192                    0.000004   \n",
        "Parvimonas            543311       0.028213                    0.002996   \n",
        "\n",
        "specific_site                 mid_vagina  palatine_tonsils  posterior_fornix  \\\n",
        "general_site                      vagina              oral            vagina   \n",
        "tax_name              tax_id                                                   \n",
        "Ruminococcus          1263       0.00000          0.006873          0.003139   \n",
        "Agrobacterium         357        0.00000          0.000000          0.000000   \n",
        "Aggregatibacter       416916     0.00000          0.050678          0.000000   \n",
        "Olsenella             133925     0.00000          0.101300          0.000000   \n",
        "Citrobacter           544        0.00000          0.000000          0.000000   \n",
        "Akkermansia           239934     0.00000          0.000000          0.000034   \n",
        "Leptotrichia          32067      0.00000          0.518070          0.000000   \n",
        "Jonquetella           428711     0.00000          0.000000          0.000000   \n",
        "Shuttleworthia        177971     0.00000          0.000000          0.000000   \n",
        "Psychrobacter         497        0.00000          0.000000          0.000000   \n",
        "Pediococcus           1253       0.00000          0.000000          0.000000   \n",
        "Phascolarctobacterium 33024      0.00000          0.000000          0.000000   \n",
        "Streptobacillus       34104      0.00000          0.000000          0.000000   \n",
        "Enterococcus          1350       0.00000          0.034695          0.000458   \n",
        "Megamonas             158846     0.00000          0.000000          0.000000   \n",
        "Lactobacillus         1578      98.55888          0.000000         92.393338   \n",
        "Sphingopyxis          165697     0.00000          0.000000          0.000076   \n",
        "Methylobacterium      407        0.00000          0.000000          0.000000   \n",
        "Brevundimonas         41275      0.00000          0.000000          0.000000   \n",
        "Delftia               80865      0.00000          0.000000          0.000000   \n",
        "Bilophila             35832      0.00000          0.000000          0.000000   \n",
        "Catenibacterium       135858     0.00000          0.000000          0.000000   \n",
        "Acidaminococcus       904        0.00000          0.000000          0.000000   \n",
        "Cardiobacterium       2717       0.00000          0.000000          0.000000   \n",
        "Solobacterium         123375     0.00000          1.004143          0.000000   \n",
        "Subdoligranulum       292632     0.00000          0.000000          0.000000   \n",
        "Lautropia             47670      0.00000          0.005002          0.000000   \n",
        "Brucella              234        0.00000          0.000000          0.000000   \n",
        "Catonella             43996      0.00000          0.069165          0.000000   \n",
        "Lactococcus           1357       0.00000          0.002952          0.000000   \n",
        "...                                  ...               ...               ...   \n",
        "Actinobacillus        713        0.00000          0.001320          0.000000   \n",
        "Leuconostoc           1243       0.00000          0.000000          0.000000   \n",
        "Holdemania            61170      0.00000          0.000000          0.000000   \n",
        "Atopobium             1380       0.00000          0.770162          0.916763   \n",
        "Proteus               583        0.00000          0.000000          0.000000   \n",
        "Marinomonas           28253      0.00000          0.000000          0.000000   \n",
        "Collinsella           102106     0.00000          0.000000          0.000000   \n",
        "Ureaplasma            2129       0.07112          0.000000          0.268771   \n",
        "Megasphaera           906        0.00000          1.748103          0.000000   \n",
        "Serratia              613        0.00000          0.000000          0.000000   \n",
        "Xanthomonas           338        0.00000          0.000000          0.000000   \n",
        "Herbaspirillum        963        0.00000          0.000000          0.000020   \n",
        "Pasteurella           745        0.00000          0.000000          0.000000   \n",
        "Micrococcus           1269       0.00000          0.000000          0.000000   \n",
        "Cellulophaga          104264     0.00000          0.002908          0.000000   \n",
        "Haemophilus           724        0.00000         13.383828          0.000000   \n",
        "Oxalobacter           846        0.00000          0.000000          0.000000   \n",
        "Alkaliphilus          114627     0.00000          0.001738          0.000000   \n",
        "Faecalibacterium      216851     0.00000          0.000082          0.000802   \n",
        "Escherichia           561        0.00000          0.000000          0.000460   \n",
        "Actinomyces           1654       0.00377          0.480800          0.000415   \n",
        "Roseburia             841        0.00000          0.000000          0.000000   \n",
        "Acidovorax            12916      0.00000          0.000000          0.000000   \n",
        "Rhodobacter           1060       0.00000          0.000000          0.000000   \n",
        "Treponema             157        0.00000          0.681318          0.000000   \n",
        "Rhodopseudomonas      1073       0.00000          0.000000          0.000000   \n",
        "Porphyromonas         836        0.00000          0.296672          0.002486   \n",
        "Parascardovia         196082     0.00000          0.000000          0.000000   \n",
        "Sutterella            40544      0.00000          0.000000          0.000000   \n",
        "Parvimonas            543311     0.00000          1.553945          0.000182   \n",
        "\n",
        "specific_site                 right_retroauricular_crease     saliva  \\\n",
        "general_site                                         skin       oral   \n",
        "tax_name              tax_id                                           \n",
        "Ruminococcus          1263                       0.000914   0.000000   \n",
        "Agrobacterium         357                        0.000008   0.000000   \n",
        "Aggregatibacter       416916                     0.000434   2.206160   \n",
        "Olsenella             133925                     0.000000   0.001173   \n",
        "Citrobacter           544                        0.000000   0.000000   \n",
        "Akkermansia           239934                     0.000000   0.000000   \n",
        "Leptotrichia          32067                      0.001141   0.185980   \n",
        "Jonquetella           428711                     0.000000   0.000000   \n",
        "Shuttleworthia        177971                     0.000000   0.014257   \n",
        "Psychrobacter         497                        0.000000   0.031707   \n",
        "Pediococcus           1253                       0.000000   0.000000   \n",
        "Phascolarctobacterium 33024                      0.000000   0.000000   \n",
        "Streptobacillus       34104                      0.000000   0.000000   \n",
        "Enterococcus          1350                       0.000067   0.041723   \n",
        "Megamonas             158846                     0.000000   0.000000   \n",
        "Lactobacillus         1578                       0.088401   0.098157   \n",
        "Sphingopyxis          165697                     0.000000   0.000000   \n",
        "Methylobacterium      407                        0.000219   0.000000   \n",
        "Brevundimonas         41275                      0.000165   0.000000   \n",
        "Delftia               80865                      0.000121   0.000000   \n",
        "Bilophila             35832                      0.000000   0.000000   \n",
        "Catenibacterium       135858                     0.000000   0.000000   \n",
        "Acidaminococcus       904                        0.000000   0.000000   \n",
        "Cardiobacterium       2717                       0.000337   0.083893   \n",
        "Solobacterium         123375                     0.000000   0.292127   \n",
        "Subdoligranulum       292632                     0.000000   0.000000   \n",
        "Lautropia             47670                      0.001015   0.387200   \n",
        "Brucella              234                        0.000000   0.000000   \n",
        "Catonella             43996                      0.000000   0.030553   \n",
        "Lactococcus           1357                       0.000726   0.003113   \n",
        "...                                                   ...        ...   \n",
        "Actinobacillus        713                        0.000000   0.007360   \n",
        "Leuconostoc           1243                       0.000000   0.005990   \n",
        "Holdemania            61170                      0.000000   0.000000   \n",
        "Atopobium             1380                       0.003071   0.970937   \n",
        "Proteus               583                        0.000000   0.000000   \n",
        "Marinomonas           28253                      0.000000   0.000000   \n",
        "Collinsella           102106                     0.000000   0.000000   \n",
        "Ureaplasma            2129                       0.000000   0.000000   \n",
        "Megasphaera           906                        0.000000   1.591690   \n",
        "Serratia              613                        0.000000   0.102950   \n",
        "Xanthomonas           338                        0.000000   0.000000   \n",
        "Herbaspirillum        963                        0.000000   0.000000   \n",
        "Pasteurella           745                        0.000036   0.000000   \n",
        "Micrococcus           1269                       0.002174   0.000000   \n",
        "Cellulophaga          104264                     0.000000   0.007527   \n",
        "Haemophilus           724                        0.076659  20.294517   \n",
        "Oxalobacter           846                        0.000000   0.000000   \n",
        "Alkaliphilus          114627                     0.000000   0.000000   \n",
        "Faecalibacterium      216851                     0.000154   0.004523   \n",
        "Escherichia           561                        0.014660   0.000000   \n",
        "Actinomyces           1654                       0.002789   1.066360   \n",
        "Roseburia             841                        0.000000   0.000000   \n",
        "Acidovorax            12916                      0.000235   0.000000   \n",
        "Rhodobacter           1060                       0.000000   0.000000   \n",
        "Treponema             157                        0.000000   0.067200   \n",
        "Rhodopseudomonas      1073                       0.000000   0.000000   \n",
        "Porphyromonas         836                        0.000682   1.260640   \n",
        "Parascardovia         196082                     0.000000   0.000000   \n",
        "Sutterella            40544                      0.000000   0.000000   \n",
        "Parvimonas            543311                     0.000000   0.208830   \n",
        "\n",
        "specific_site                    stool  subgingival_plaque  \\\n",
        "general_site                     stool                oral   \n",
        "tax_name              tax_id                                 \n",
        "Ruminococcus          1263    3.701499            0.000230   \n",
        "Agrobacterium         357     0.000000            0.000000   \n",
        "Aggregatibacter       416916  0.000107            2.010644   \n",
        "Olsenella             133925  0.000000            0.056920   \n",
        "Citrobacter           544     0.000883            0.000000   \n",
        "Akkermansia           239934  0.899293            0.000000   \n",
        "Leptotrichia          32067   0.000024            1.960361   \n",
        "Jonquetella           428711  0.000000            0.000000   \n",
        "Shuttleworthia        177971  0.000000            0.029916   \n",
        "Psychrobacter         497     0.000000            0.000000   \n",
        "Pediococcus           1253    0.000030            0.000000   \n",
        "Phascolarctobacterium 33024   0.142882            0.000000   \n",
        "Streptobacillus       34104   0.000000            0.000000   \n",
        "Enterococcus          1350    0.007888            0.004329   \n",
        "Megamonas             158846  0.042272            0.000000   \n",
        "Lactobacillus         1578    0.012162            0.000913   \n",
        "Sphingopyxis          165697  0.000000            0.000000   \n",
        "Methylobacterium      407     0.000209            0.000000   \n",
        "Brevundimonas         41275   0.000000            0.000000   \n",
        "Delftia               80865   0.000000            0.000000   \n",
        "Bilophila             35832   0.065139            0.000000   \n",
        "Catenibacterium       135858  0.056663            0.000000   \n",
        "Acidaminococcus       904     0.000042            0.000000   \n",
        "Cardiobacterium       2717    0.000001            2.559864   \n",
        "Solobacterium         123375  0.001283            0.103901   \n",
        "Subdoligranulum       292632  0.015273            0.000000   \n",
        "Lautropia             47670   0.000000            0.954963   \n",
        "Brucella              234     0.000014            0.000000   \n",
        "Catonella             43996   0.000000            0.330303   \n",
        "Lactococcus           1357    0.011050            0.000000   \n",
        "...                                ...                 ...   \n",
        "Actinobacillus        713     0.000000            0.000000   \n",
        "Leuconostoc           1243    0.000227            0.000000   \n",
        "Holdemania            61170   0.026074            0.000000   \n",
        "Atopobium             1380    0.000291            0.261836   \n",
        "Proteus               583     0.000000            0.000000   \n",
        "Marinomonas           28253   0.000000            0.000000   \n",
        "Collinsella           102106  0.206767            0.000000   \n",
        "Ureaplasma            2129    0.000000            0.000000   \n",
        "Megasphaera           906     0.003123            0.052337   \n",
        "Serratia              613     0.000000            0.000000   \n",
        "Xanthomonas           338     0.000000            0.000000   \n",
        "Herbaspirillum        963     0.000000            0.000000   \n",
        "Pasteurella           745     0.000000            0.000000   \n",
        "Micrococcus           1269    0.000000            0.000000   \n",
        "Cellulophaga          104264  0.000077            0.046329   \n",
        "Haemophilus           724     0.088559            2.898726   \n",
        "Oxalobacter           846     0.008908            0.000000   \n",
        "Alkaliphilus          114627  0.000587            0.000444   \n",
        "Faecalibacterium      216851  3.482948            0.000000   \n",
        "Escherichia           561     0.809806            0.321337   \n",
        "Actinomyces           1654    0.000181            3.418323   \n",
        "Roseburia             841     1.451654            0.000000   \n",
        "Acidovorax            12916   0.000000            0.000000   \n",
        "Rhodobacter           1060    0.000000            0.000000   \n",
        "Treponema             157     0.000000            3.899533   \n",
        "Rhodopseudomonas      1073    0.000009            0.000000   \n",
        "Porphyromonas         836     0.005276            1.862417   \n",
        "Parascardovia         196082  0.000000            0.000000   \n",
        "Sutterella            40544   1.099559            0.000000   \n",
        "Parvimonas            543311  0.001054            2.574669   \n",
        "\n",
        "specific_site                 supragingival_plaque     throat  tongue_dorsum  \\\n",
        "general_site                                  oral     throat         throat   \n",
        "tax_name              tax_id                                                   \n",
        "Ruminococcus          1263                0.001331   0.000000       0.000410   \n",
        "Agrobacterium         357                 0.000000   0.000000       0.000000   \n",
        "Aggregatibacter       416916              2.338830   0.030823       0.063490   \n",
        "Olsenella             133925              0.002011   0.008174       0.000377   \n",
        "Citrobacter           544                 0.000000   0.000000       0.000000   \n",
        "Akkermansia           239934              0.000012   0.000000       0.000006   \n",
        "Leptotrichia          32067               2.123331   0.304966       0.573608   \n",
        "Jonquetella           428711              0.000000   0.000000       0.000000   \n",
        "Shuttleworthia        177971              0.002324   0.000353       0.000428   \n",
        "Psychrobacter         497                 0.000000   0.000000       0.000005   \n",
        "Pediococcus           1253                0.000000   0.000000       0.000000   \n",
        "Phascolarctobacterium 33024               0.000000   0.000000       0.000000   \n",
        "Streptobacillus       34104               0.000002   0.000000       0.000002   \n",
        "Enterococcus          1350                0.007862   0.014853       0.035602   \n",
        "Megamonas             158846              0.000519   0.000000       0.000140   \n",
        "Lactobacillus         1578                0.001525   0.000457       0.000708   \n",
        "Sphingopyxis          165697              0.000000   0.000000       0.000000   \n",
        "Methylobacterium      407                 0.000114   0.000000       0.000030   \n",
        "Brevundimonas         41275               0.000000   0.000000       0.000000   \n",
        "Delftia               80865               0.000000   0.000000       0.000000   \n",
        "Bilophila             35832               0.000005   0.000000       0.000000   \n",
        "Catenibacterium       135858              0.000000   0.000000       0.000000   \n",
        "Acidaminococcus       904                 0.000000   0.000000       0.000000   \n",
        "Cardiobacterium       2717                1.420594   0.002666       0.023617   \n",
        "Solobacterium         123375              0.030559   0.770401       0.862904   \n",
        "Subdoligranulum       292632              0.000000   0.000000       0.000000   \n",
        "Lautropia             47670               2.685566   0.018849       0.098337   \n",
        "Brucella              234                 0.000000   0.000000       0.000000   \n",
        "Catonella             43996               0.192797   0.030833       0.023674   \n",
        "Lactococcus           1357                0.001020   0.000000       0.001197   \n",
        "...                                            ...        ...            ...   \n",
        "Actinobacillus        713                 0.000056   0.000000       0.000104   \n",
        "Leuconostoc           1243                0.000000   0.000000       0.000010   \n",
        "Holdemania            61170               0.000000   0.000000       0.000000   \n",
        "Atopobium             1380                0.062533   0.429936       0.687797   \n",
        "Proteus               583                 0.000000   0.000000       0.000000   \n",
        "Marinomonas           28253               0.000078   0.000000       0.000029   \n",
        "Collinsella           102106              0.000000   0.000000       0.000000   \n",
        "Ureaplasma            2129                0.000000   0.000000       0.000014   \n",
        "Megasphaera           906                 0.008704   1.944736       1.173358   \n",
        "Serratia              613                 0.000000   0.000000       0.000002   \n",
        "Xanthomonas           338                 0.000000   0.000000       0.000012   \n",
        "Herbaspirillum        963                 0.000000   0.000000       0.000000   \n",
        "Pasteurella           745                 0.000109   0.000000       0.000016   \n",
        "Micrococcus           1269                0.000000   0.000000       0.000000   \n",
        "Cellulophaga          104264              0.049064   0.013480       0.007234   \n",
        "Haemophilus           724                 7.760013  10.312489      18.012251   \n",
        "Oxalobacter           846                 0.000000   0.000000       0.000000   \n",
        "Alkaliphilus          114627              0.000229   0.003694       0.004875   \n",
        "Faecalibacterium      216851              0.000017   0.000000       0.000092   \n",
        "Escherichia           561                 0.000210   0.000766       0.000165   \n",
        "Actinomyces           1654                6.256130   0.583140       1.645602   \n",
        "Roseburia             841                 0.000000   0.000000       0.000005   \n",
        "Acidovorax            12916               0.000262   0.000000       0.000022   \n",
        "Rhodobacter           1060                0.000023   0.000000       0.000000   \n",
        "Treponema             157                 0.344508   0.006640       0.025036   \n",
        "Rhodopseudomonas      1073                0.000000   0.000000       0.000000   \n",
        "Porphyromonas         836                 0.322816   0.024234       0.050447   \n",
        "Parascardovia         196082              0.000000   0.000000       0.000003   \n",
        "Sutterella            40544               0.000001   0.000000       0.000032   \n",
        "Parvimonas            543311              0.126719   0.305444       0.149829   \n",
        "\n",
        "specific_site                 vaginal_introitus  \n",
        "general_site                             vagina  \n",
        "tax_name              tax_id                     \n",
        "Ruminococcus          1263             0.000000  \n",
        "Agrobacterium         357              0.000000  \n",
        "Aggregatibacter       416916           0.000000  \n",
        "Olsenella             133925           0.000000  \n",
        "Citrobacter           544              0.000000  \n",
        "Akkermansia           239934           0.000000  \n",
        "Leptotrichia          32067            0.000000  \n",
        "Jonquetella           428711           0.000000  \n",
        "Shuttleworthia        177971           0.000000  \n",
        "Psychrobacter         497              0.000000  \n",
        "Pediococcus           1253             0.000000  \n",
        "Phascolarctobacterium 33024            0.000000  \n",
        "Streptobacillus       34104            0.000000  \n",
        "Enterococcus          1350             0.000000  \n",
        "Megamonas             158846           0.000000  \n",
        "Lactobacillus         1578            88.898777  \n",
        "Sphingopyxis          165697           0.000000  \n",
        "Methylobacterium      407              0.000000  \n",
        "Brevundimonas         41275            0.000000  \n",
        "Delftia               80865            0.000000  \n",
        "Bilophila             35832            0.000000  \n",
        "Catenibacterium       135858           0.000000  \n",
        "Acidaminococcus       904              0.000000  \n",
        "Cardiobacterium       2717             0.000000  \n",
        "Solobacterium         123375           0.000000  \n",
        "Subdoligranulum       292632           0.000000  \n",
        "Lautropia             47670            0.000000  \n",
        "Brucella              234              0.000000  \n",
        "Catonella             43996            0.000000  \n",
        "Lactococcus           1357             0.000000  \n",
        "...                                         ...  \n",
        "Actinobacillus        713              0.000000  \n",
        "Leuconostoc           1243             0.000000  \n",
        "Holdemania            61170            0.000000  \n",
        "Atopobium             1380             0.000000  \n",
        "Proteus               583              0.000000  \n",
        "Marinomonas           28253            0.000000  \n",
        "Collinsella           102106           0.000000  \n",
        "Ureaplasma            2129             1.389137  \n",
        "Megasphaera           906              0.000000  \n",
        "Serratia              613              0.000000  \n",
        "Xanthomonas           338              0.000000  \n",
        "Herbaspirillum        963              0.008177  \n",
        "Pasteurella           745              0.000000  \n",
        "Micrococcus           1269             0.000000  \n",
        "Cellulophaga          104264           0.000000  \n",
        "Haemophilus           724              0.000413  \n",
        "Oxalobacter           846              0.000000  \n",
        "Alkaliphilus          114627           0.000000  \n",
        "Faecalibacterium      216851           0.000000  \n",
        "Escherichia           561              0.000000  \n",
        "Actinomyces           1654             0.000000  \n",
        "Roseburia             841              0.000000  \n",
        "Acidovorax            12916            0.000000  \n",
        "Rhodobacter           1060             0.000000  \n",
        "Treponema             157              0.000000  \n",
        "Rhodopseudomonas      1073             0.000000  \n",
        "Porphyromonas         836              0.000000  \n",
        "Parascardovia         196082           0.000000  \n",
        "Sutterella            40544            0.000000  \n",
        "Parvimonas            543311           0.000000  \n",
        "\n",
        "[139 rows x 15 columns]"
       ]
      }
     ],
     "prompt_number": 110
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}